{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ACCESS BIOINFORMATICS DATABASES WITH BIO-PYTHON"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. [NCBI](#1.-NCBI)<br>\n",
    "    1.1. [Nucleotide BLAST](#1.1.-Nucleotide-BLAST)<br>\n",
    "    1.2. [Protein BLAST](#1.2.-Protein-BLAST)\n",
    "    \n",
    "2. [ENTREZ](#2.-ENTREZ)<br>\n",
    "    2.1. [PUBMED](#2.1.-PUBMED)<br>\n",
    "    2.2. [Nucleotide](#2.2.-Nucleotide)\n",
    "    \n",
    "3. [PDB](#3.-PDB)\n",
    "\n",
    "4. [EXPASY](#4.-EXPASY)<br>\n",
    "    4.1. [PROSITE](#4.1.-PROSITE)<br>\n",
    "    4.2. [ScanProsite](#4.2.-ScanProsite)\n",
    "    \n",
    "5. [KEGG](#5.-KEGG)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIWWW\n",
    "from Bio import SeqIO, SearchIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on module Bio.Blast.NCBIWWW in Bio.Blast:\n",
      "\n",
      "NAME\n",
      "    Bio.Blast.NCBIWWW - Code to invoke the NCBI BLAST server over the internet.\n",
      "\n",
      "DESCRIPTION\n",
      "    This module provides code to work with the WWW version of BLAST\n",
      "    provided by the NCBI. https://blast.ncbi.nlm.nih.gov/\n",
      "\n",
      "FUNCTIONS\n",
      "    qblast(program, database, sequence, url_base='https://blast.ncbi.nlm.nih.gov/Blast.cgi', auto_format=None, composition_based_statistics=None, db_genetic_code=None, endpoints=None, entrez_query='(none)', expect=10.0, filter=None, gapcosts=None, genetic_code=None, hitlist_size=50, i_thresh=None, layout=None, lcase_mask=None, matrix_name=None, nucl_penalty=None, nucl_reward=None, other_advanced=None, perc_ident=None, phi_pattern=None, query_file=None, query_believe_defline=None, query_from=None, query_to=None, searchsp_eff=None, service=None, threshold=None, ungapped_alignment=None, word_size=None, short_query=None, alignments=500, alignment_view=None, descriptions=500, entrez_links_new_window=None, expect_low=None, expect_high=None, format_entrez_query=None, format_object=None, format_type='XML', ncbi_gi=None, results_file=None, show_overview=None, megablast=None, template_type=None, template_length=None)\n",
      "        BLAST search using NCBI's QBLAST server or a cloud service provider.\n",
      "        \n",
      "        Supports all parameters of the old qblast API for Put and Get.\n",
      "        \n",
      "        Please note that NCBI uses the new Common URL API for BLAST searches\n",
      "        on the internet (http://ncbi.github.io/blast-cloud/dev/api.html). Thus,\n",
      "        some of the parameters used by this function are not (or are no longer)\n",
      "        officially supported by NCBI. Although they are still functioning, this\n",
      "        may change in the future.\n",
      "        \n",
      "        The Common URL API (http://ncbi.github.io/blast-cloud/dev/api.html) allows\n",
      "        doing BLAST searches on cloud servers. To use this feature, please set\n",
      "        ``url_base='http://host.my.cloud.service.provider.com/cgi-bin/blast.cgi'``\n",
      "        and ``format_object='Alignment'``. For more details, please see\n",
      "        https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&DOC_TYPE=CloudBlast\n",
      "        \n",
      "        Some useful parameters:\n",
      "        \n",
      "         - program        blastn, blastp, blastx, tblastn, or tblastx (lower case)\n",
      "         - database       Which database to search against (e.g. \"nr\").\n",
      "         - sequence       The sequence to search.\n",
      "         - ncbi_gi        TRUE/FALSE whether to give 'gi' identifier.\n",
      "         - descriptions   Number of descriptions to show.  Def 500.\n",
      "         - alignments     Number of alignments to show.  Def 500.\n",
      "         - expect         An expect value cutoff.  Def 10.0.\n",
      "         - matrix_name    Specify an alt. matrix (PAM30, PAM70, BLOSUM80, BLOSUM45).\n",
      "         - filter         \"none\" turns off filtering.  Default no filtering\n",
      "         - format_type    \"HTML\", \"Text\", \"ASN.1\", or \"XML\".  Def. \"XML\".\n",
      "         - entrez_query   Entrez query to limit Blast search\n",
      "         - hitlist_size   Number of hits to return. Default 50\n",
      "         - megablast      TRUE/FALSE whether to use MEga BLAST algorithm (blastn only)\n",
      "         - short_query    TRUE/FALSE whether to adjust the search parameters for a\n",
      "                          short query sequence. Note that this will override\n",
      "                          manually set parameters like word size and e value. Turns\n",
      "                          off when sequence length is > 30 residues. Default: None.\n",
      "         - service        plain, psi, phi, rpsblast, megablast (lower case)\n",
      "        \n",
      "        This function does no checking of the validity of the parameters\n",
      "        and passes the values to the server as is.  More help is available at:\n",
      "        https://ncbi.github.io/blast-cloud/dev/api.html\n",
      "\n",
      "DATA\n",
      "    NCBI_BLAST_URL = 'https://blast.ncbi.nlm.nih.gov/Blast.cgi'\n",
      "\n",
      "FILE\n",
      "    /home/rhyme/.local/lib/python3.6/site-packages/Bio/Blast/NCBIWWW.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#help(NCBIWWW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.1. Nucleotide BLAST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dir  notebook.ipynb  nuc_seq.fasta  obsolete  prot_seq.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "774"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nuc_record = SeqIO.read(\"nuc_seq.fasta\", format = \"fasta\")\n",
    "len(nuc_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", nuc_record.seq)\n",
    "blast_result = SearchIO.read(result_handle,\"blast-xml\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Program: blastn (2.14.1+)\n",
      "  Query: No (774)\n",
      "         definition line\n",
      " Target: nt\n",
      "   Hits: ----  -----  ----------------------------------------------------------\n",
      "            #  # HSP  ID + description\n",
      "         ----  -----  ----------------------------------------------------------\n",
      "            0      1  gi|2621968847|emb|OY783932.1|  Severe acute respiratory...\n",
      "            1      1  gi|2621965820|emb|OY785339.1|  Severe acute respiratory...\n"
     ]
    }
   ],
   "source": [
    "print(blast_result[0:2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequence ID: gi|2621968847|emb|OY783932.1|\n",
      "Sequence Description: Severe acute respiratory syndrome coronavirus 2 genome assembly, chromosome: 1\n",
      "E-value: 0.0\n"
     ]
    }
   ],
   "source": [
    "Seq = blast_result[0]\n",
    "print(f\"Sequence ID: {Seq.id}\")\n",
    "print(f\"Sequence Description: {Seq.description}\")\n",
    "\n",
    "details = Seq[0]\n",
    "print(f\"E-value: {details.evalue}\")\n",
    "\n",
    "#E_value is 0.0 sequence is exaxt or very closely match to our sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "alignment:\n",
      "Alignment with 2 rows and 774 columns\n",
      "ATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAA...GGT No\n",
      "ATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAA...GGT gi|2621968847|emb|OY783932.1|\n"
     ]
    }
   ],
   "source": [
    "print(f\"alignment:\\n{details.aln}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.2. Protein BLAST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "258"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_record = SeqIO.read(\"prot_seq.fasta\", format=\"fasta\")\n",
    "len(prot_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle = NCBIWWW.qblast(\"blastp\", \"pdb\", prot_record.seq)\n",
    "blast_result = SearchIO.read(result_handle, \"blast-xml\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Program: blastp (2.14.1+)\n",
      "  Query: unnamed (258)\n",
      "         protein product\n",
      " Target: pdb\n",
      "   Hits: ----  -----  ----------------------------------------------------------\n",
      "            #  # HSP  ID + description\n",
      "         ----  -----  ----------------------------------------------------------\n",
      "            0      1  pdb|8ELJ|A  Chain A, Spike glycoprotein [Severe acute r...\n",
      "            1      1  pdb|7CAB|A  Chain A, Spike glycoprotein [Severe acute r...\n"
     ]
    }
   ],
   "source": [
    "print(blast_result[0:2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequence ID: pdb|8ELJ|A\n",
      "Sequence Description: Chain A, Spike glycoprotein [Severe acute respiratory syndrome coronavirus 2]\n",
      "E-value: 0.0\n"
     ]
    }
   ],
   "source": [
    "Seq = blast_result [0]\n",
    "print(f\"Sequence ID: {Seq.id}\")\n",
    "print(f\"Sequence Description: {Seq.description}\")\n",
    "\n",
    "details = Seq[0]\n",
    "print(f\"E-value: {details.evalue}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "alignment:\n",
      " Alignment with 2 rows and 258 columns\n",
      "IAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNYLY...PIG unnamed\n",
      "IAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNYLY...PIG pdb|8ELJ|A\n"
     ]
    }
   ],
   "source": [
    "print(f\"alignment:\\n {details.aln}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. ENTREZ"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on package Bio.Entrez in Bio:\n",
      "\n",
      "NAME\n",
      "    Bio.Entrez - Provides code to access NCBI over the WWW.\n",
      "\n",
      "DESCRIPTION\n",
      "    The main Entrez web page is available at:\n",
      "    http://www.ncbi.nlm.nih.gov/Entrez/\n",
      "    \n",
      "    Entrez Programming Utilities web page is available at:\n",
      "    http://www.ncbi.nlm.nih.gov/books/NBK25501/\n",
      "    \n",
      "    This module provides a number of functions like ``efetch`` (short for\n",
      "    Entrez Fetch) which will return the data as a handle object. This is\n",
      "    a standard interface used in Python for reading data from a file, or\n",
      "    in this case a remote network connection, and provides methods like\n",
      "    ``.read()`` or offers iteration over the contents line by line. See\n",
      "    also \"What the heck is a handle?\" in the Biopython Tutorial and\n",
      "    Cookbook: http://biopython.org/DIST/docs/tutorial/Tutorial.html\n",
      "    http://biopython.org/DIST/docs/tutorial/Tutorial.pdf\n",
      "    The handle returned by these functions can be either in text mode or\n",
      "    in binary mode, depending on the data requested and the results\n",
      "    returned by NCBI Entrez. Typically, XML data will be in binary mode\n",
      "    while other data will be in text mode, as required by the downstream\n",
      "    parser to parse the data.\n",
      "    \n",
      "    Unlike a handle to a file on disk from the ``open(filename)`` function,\n",
      "    which has a ``.name`` attribute giving the filename, the handles from\n",
      "    ``Bio.Entrez`` all have a ``.url`` attribute instead giving the URL\n",
      "    used to connect to the NCBI Entrez API.\n",
      "    \n",
      "    All the functions that send requests to the NCBI Entrez API will\n",
      "    automatically respect the NCBI rate limit (of 3 requests per second\n",
      "    without an API key, or 10 requests per second with an API key) and\n",
      "    will automatically retry when encountering transient failures\n",
      "    (i.e. connection failures or HTTP 5XX codes). By default, Biopython\n",
      "    does a maximum of three tries before giving up, and sleeps for 15\n",
      "    seconds between tries. You can tweak these parameters by setting\n",
      "    ``Bio.Entrez.max_tries`` and ``Bio.Entrez.sleep_between_tries``.\n",
      "    \n",
      "    The Entrez module also provides an XML parser which takes a handle\n",
      "    as input.\n",
      "    \n",
      "    Variables:\n",
      "    \n",
      "        - email        Set the Entrez email parameter (default is not set).\n",
      "        - tool         Set the Entrez tool parameter (default is ``biopython``).\n",
      "        - api_key      Personal API key from NCBI. If not set, only 3 queries per\n",
      "          second are allowed. 10 queries per seconds otherwise with a\n",
      "          valid API key.\n",
      "        - max_tries    Configures how many times failed requests will be\n",
      "          automatically retried on error (default is 3).\n",
      "        - sleep_between_tries   The delay, in seconds, before retrying a request on\n",
      "          error (default is 15).\n",
      "    \n",
      "    Functions:\n",
      "    \n",
      "        - efetch       Retrieves records in the requested format from a list of one or\n",
      "          more primary IDs or from the user's environment\n",
      "        - epost        Posts a file containing a list of primary IDs for future use in\n",
      "          the user's environment to use with subsequent search strategies\n",
      "        - esearch      Searches and retrieves primary IDs (for use in EFetch, ELink,\n",
      "          and ESummary) and term translations and optionally retains\n",
      "          results for future use in the user's environment.\n",
      "        - elink        Checks for the existence of an external or Related Articles link\n",
      "          from a list of one or more primary IDs.  Retrieves primary IDs\n",
      "          and relevancy scores for links to Entrez databases or Related\n",
      "          Articles;  creates a hyperlink to the primary LinkOut provider\n",
      "          for a specific ID and database, or lists LinkOut URLs\n",
      "          and Attributes for multiple IDs.\n",
      "        - einfo        Provides field index term counts, last update, and available\n",
      "          links for each database.\n",
      "        - esummary     Retrieves document summaries from a list of primary IDs or from\n",
      "          the user's environment.\n",
      "        - egquery      Provides Entrez database counts in XML for a single search\n",
      "          using Global Query.\n",
      "        - espell       Retrieves spelling suggestions.\n",
      "        - ecitmatch    Retrieves PubMed IDs (PMIDs) that correspond to a set of\n",
      "          input citation strings.\n",
      "    \n",
      "        - read         Parses the XML results returned by any of the above functions.\n",
      "          Alternatively, the XML data can be read from a file opened in binary mode.\n",
      "          Typical usage is:\n",
      "    \n",
      "              >>> from Bio import Entrez\n",
      "              >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "              >>> handle = Entrez.einfo() # or esearch, efetch, ...\n",
      "              >>> record = Entrez.read(handle)\n",
      "              >>> handle.close()\n",
      "    \n",
      "           where record is now a Python dictionary or list.\n",
      "    \n",
      "        - parse        Parses the XML results returned by those of the above functions\n",
      "          which can return multiple records - such as efetch, esummary\n",
      "          and elink. Typical usage is:\n",
      "    \n",
      "              >>> handle = Entrez.esummary(db=\"pubmed\", id=\"19304878,14630660\", retmode=\"xml\")\n",
      "              >>> records = Entrez.parse(handle)\n",
      "              >>> for record in records:\n",
      "              ...     # each record is a Python dictionary or list.\n",
      "              ...     print(record['Title'])\n",
      "              Biopython: freely available Python tools for computational molecular biology and bioinformatics.\n",
      "              PDB file parser and structure class implemented in Python.\n",
      "              >>> handle.close()\n",
      "    \n",
      "          This function is appropriate only if the XML file contains\n",
      "          multiple records, and is particular useful for large files.\n",
      "    \n",
      "        - _open        Internally used function.\n",
      "\n",
      "PACKAGE CONTENTS\n",
      "    Parser\n",
      "\n",
      "FUNCTIONS\n",
      "    ecitmatch(**keywds)\n",
      "        Retrieve PMIDs for input citation strings, returned as a handle.\n",
      "        \n",
      "        ECitMatch retrieves PubMed IDs (PMIDs) that correspond to a set of input\n",
      "        citation strings.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ECitMatch\n",
      "        \n",
      "        Return a handle to the results, by default in plain text\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> citation_1 = {\"journal_title\": \"proc natl acad sci u s a\",\n",
      "        ...               \"year\": \"1991\", \"volume\": \"88\", \"first_page\": \"3248\",\n",
      "        ...               \"author_name\": \"mann bj\", \"key\": \"citation_1\"}\n",
      "        >>> handle = Entrez.ecitmatch(db=\"pubmed\", bdata=[citation_1])\n",
      "        >>> print(handle.read().strip().split(\"|\"))\n",
      "        ['proc natl acad sci u s a', '1991', '88', '3248', 'mann bj', 'citation_1', '2014248']\n",
      "        >>> handle.close()\n",
      "    \n",
      "    efetch(db, **keywords)\n",
      "        Fetch Entrez results which are returned as a handle.\n",
      "        \n",
      "        EFetch retrieves records in the requested format from a list or set of one or\n",
      "        more UIs or from user's environment.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch\n",
      "        \n",
      "        Return a handle to the results.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.efetch(db=\"nucleotide\", id=\"AY851612\", rettype=\"gb\", retmode=\"text\")\n",
      "        >>> print(handle.readline().strip())\n",
      "        LOCUS       AY851612                 892 bp    DNA     linear   PLN 10-APR-2007\n",
      "        >>> handle.close()\n",
      "        \n",
      "        This will automatically use an HTTP POST rather than HTTP GET if there\n",
      "        are over 200 identifiers as recommended by the NCBI.\n",
      "        \n",
      "        **Warning:** The NCBI changed the default retmode in Feb 2012, so many\n",
      "        databases which previously returned text output now give XML.\n",
      "    \n",
      "    egquery(**keywds)\n",
      "        Provide Entrez database counts for a global search.\n",
      "        \n",
      "        EGQuery provides Entrez database counts in XML for a single search\n",
      "        using Global Query.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EGQuery\n",
      "        \n",
      "        Return a handle to the results in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        This quick example based on a longer version from the Biopython\n",
      "        Tutorial just checks there are over 60 matches for 'Biopython'\n",
      "        in PubMedCentral:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.egquery(term=\"biopython\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> for row in record[\"eGQueryResult\"]:\n",
      "        ...     if \"pmc\" in row[\"DbName\"]:\n",
      "        ...         print(int(row[\"Count\"]) > 60)\n",
      "        True\n",
      "    \n",
      "    einfo(**keywds)\n",
      "        Return a summary of the Entrez databases as a results handle.\n",
      "        \n",
      "        EInfo provides field names, index term counts, last update, and\n",
      "        available links for each Entrez database.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EInfo\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> record = Entrez.read(Entrez.einfo())\n",
      "        >>> 'pubmed' in record['DbList']\n",
      "        True\n",
      "    \n",
      "    elink(**keywds)\n",
      "        Check for linked external articles and return a handle.\n",
      "        \n",
      "        ELink checks for the existence of an external or Related Articles link\n",
      "        from a list of one or more primary IDs;  retrieves IDs and relevancy\n",
      "        scores for links to Entrez databases or Related Articles; creates a\n",
      "        hyperlink to the primary LinkOut provider for a specific ID and\n",
      "        database, or lists LinkOut URLs and attributes for multiple IDs.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ELink\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        This example finds articles related to the Biopython application\n",
      "        note's entry in the PubMed database:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> pmid = \"19304878\"\n",
      "        >>> handle = Entrez.elink(dbfrom=\"pubmed\", id=pmid, linkname=\"pubmed_pubmed\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> print(record[0][\"LinkSetDb\"][0][\"LinkName\"])\n",
      "        pubmed_pubmed\n",
      "        >>> linked = [link[\"Id\"] for link in record[0][\"LinkSetDb\"][0][\"Link\"]]\n",
      "        >>> \"17121776\" in linked\n",
      "        True\n",
      "        \n",
      "        This is explained in much more detail in the Biopython Tutorial.\n",
      "    \n",
      "    epost(db, **keywds)\n",
      "        Post a file of identifiers for future use.\n",
      "        \n",
      "        Posts a file containing a list of UIs for future use in the user's\n",
      "        environment to use with subsequent search strategies.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EPost\n",
      "        \n",
      "        Return a handle to the results.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "    \n",
      "    esearch(db, term, **keywds)\n",
      "        Run an Entrez search and return a handle to the results.\n",
      "        \n",
      "        ESearch searches and retrieves primary IDs (for use in EFetch, ELink\n",
      "        and ESummary) and term translations, and optionally retains results\n",
      "        for future use in the user's environment.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESearch\n",
      "        \n",
      "        Return a handle to the results which are always in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.esearch(db=\"nucleotide\", retmax=10, term=\"opuntia[ORGN] accD\", idtype=\"acc\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> int(record[\"Count\"]) >= 2\n",
      "        True\n",
      "        >>> \"EF590893.1\" in record[\"IdList\"]\n",
      "        True\n",
      "        >>> \"EF590892.1\" in record[\"IdList\"]\n",
      "        True\n",
      "    \n",
      "    espell(**keywds)\n",
      "        Retrieve spelling suggestions as a results handle.\n",
      "        \n",
      "        ESpell retrieves spelling suggestions, if available.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESpell\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> record = Entrez.read(Entrez.espell(term=\"biopythooon\"))\n",
      "        >>> print(record[\"Query\"])\n",
      "        biopythooon\n",
      "        >>> print(record[\"CorrectedQuery\"])\n",
      "        biopython\n",
      "    \n",
      "    esummary(**keywds)\n",
      "        Retrieve document summaries as a results handle.\n",
      "        \n",
      "        ESummary retrieves document summaries from a list of primary IDs or\n",
      "        from the user's environment.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESummary\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        This example discovers more about entry 19923 in the structure\n",
      "        database:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.esummary(db=\"structure\", id=\"19923\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> print(record[0][\"Id\"])\n",
      "        19923\n",
      "        >>> print(record[0][\"PdbDescr\"])\n",
      "        Crystal Structure Of E. Coli Aconitase B\n",
      "    \n",
      "    parse(handle, validate=True, escape=False)\n",
      "        Parse an XML file from the NCBI Entrez Utilities into python objects.\n",
      "        \n",
      "        This function parses an XML file created by NCBI's Entrez Utilities,\n",
      "        returning a multilevel data structure of Python lists and dictionaries.\n",
      "        This function is suitable for XML files that (in Python) can be represented\n",
      "        as a list of individual records. Whereas 'read' reads the complete file\n",
      "        and returns a single Python list, 'parse' is a generator function that\n",
      "        returns the records one by one. This function is therefore particularly\n",
      "        useful for parsing large files.\n",
      "        \n",
      "        Most XML files returned by NCBI's Entrez Utilities can be parsed by\n",
      "        this function, provided its DTD is available. Biopython includes the\n",
      "        DTDs for most commonly used Entrez Utilities.\n",
      "        \n",
      "        The handle must be in binary mode. This allows the parser to detect the\n",
      "        encoding from the XML file, and to use it to convert all text in the XML\n",
      "        to the correct Unicode string. The functions in Bio.Entrez to access NCBI\n",
      "        Entrez will automatically return XML data in binary mode. For files,\n",
      "        please use mode \"rb\" when opening the file, as in\n",
      "        \n",
      "            >>> from Bio import Entrez\n",
      "            >>> handle = open(\"Entrez/pubmed1.xml\", \"rb\")  # opened in binary mode\n",
      "            >>> records = Entrez.parse(handle)\n",
      "            >>> for record in records:\n",
      "            ...     print(record['MedlineCitation']['Article']['Journal']['Title'])\n",
      "            ...\n",
      "            Social justice (San Francisco, Calif.)\n",
      "            Biochimica et biophysica acta\n",
      "            >>> handle.close()\n",
      "        \n",
      "        If validate is True (default), the parser will validate the XML file\n",
      "        against the DTD, and raise an error if the XML file contains tags that\n",
      "        are not represented in the DTD. If validate is False, the parser will\n",
      "        simply skip such tags.\n",
      "        \n",
      "        If escape is True, all characters that are not valid HTML are replaced\n",
      "        by HTML escape characters to guarantee that the returned strings are\n",
      "        valid HTML fragments. For example, a less-than sign (<) is replaced by\n",
      "        &lt;. If escape is False (default), the string is returned as is.\n",
      "        \n",
      "        Whereas the data structure seems to consist of generic Python lists,\n",
      "        dictionaries, strings, and so on, each of these is actually a class\n",
      "        derived from the base type. This allows us to store the attributes\n",
      "        (if any) of each element in a dictionary my_element.attributes, and\n",
      "        the tag name in my_element.tag.\n",
      "    \n",
      "    read(handle, validate=True, escape=False)\n",
      "        Parse an XML file from the NCBI Entrez Utilities into python objects.\n",
      "        \n",
      "        This function parses an XML file created by NCBI's Entrez Utilities,\n",
      "        returning a multilevel data structure of Python lists and dictionaries.\n",
      "        Most XML files returned by NCBI's Entrez Utilities can be parsed by\n",
      "        this function, provided its DTD is available. Biopython includes the\n",
      "        DTDs for most commonly used Entrez Utilities.\n",
      "        \n",
      "        The handle must be in binary mode. This allows the parser to detect the\n",
      "        encoding from the XML file, and to use it to convert all text in the XML\n",
      "        to the correct Unicode string. The functions in Bio.Entrez to access NCBI\n",
      "        Entrez will automatically return XML data in binary mode. For files,\n",
      "        please use mode \"rb\" when opening the file, as in\n",
      "        \n",
      "            >>> from Bio import Entrez\n",
      "            >>> handle = open(\"Entrez/esearch1.xml\", \"rb\")  # opened in binary mode\n",
      "            >>> record = Entrez.read(handle)\n",
      "            >>> print(record['QueryTranslation'])\n",
      "            biopython[All Fields]\n",
      "            >>> handle.close()\n",
      "        \n",
      "        If validate is True (default), the parser will validate the XML file\n",
      "        against the DTD, and raise an error if the XML file contains tags that\n",
      "        are not represented in the DTD. If validate is False, the parser will\n",
      "        simply skip such tags.\n",
      "        \n",
      "        If escape is True, all characters that are not valid HTML are replaced\n",
      "        by HTML escape characters to guarantee that the returned strings are\n",
      "        valid HTML fragments. For example, a less-than sign (<) is replaced by\n",
      "        &lt;. If escape is False (default), the string is returned as is.\n",
      "        \n",
      "        Whereas the data structure seems to consist of generic Python lists,\n",
      "        dictionaries, strings, and so on, each of these is actually a class\n",
      "        derived from the base type. This allows us to store the attributes\n",
      "        (if any) of each element in a dictionary my_element.attributes, and\n",
      "        the tag name in my_element.tag.\n",
      "\n",
      "DATA\n",
      "    api_key = None\n",
      "    email = None\n",
      "    max_tries = 3\n",
      "    sleep_between_tries = 15\n",
      "    tool = 'biopython'\n",
      "\n",
      "FILE\n",
      "    /home/rhyme/.local/lib/python3.6/site-packages/Bio/Entrez/__init__.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(Entrez)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "Entrez.email = \"datacyclopes@gmail.com\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['pubmed', 'protein', 'nuccore', 'ipg', 'nucleotide', 'structure', 'genome', 'annotinfo', 'assembly', 'bioproject', 'biosample', 'blastdbinfo', 'books', 'cdd', 'clinvar', 'gap', 'gapplus', 'grasp', 'dbvar', 'gene', 'gds', 'geoprofiles', 'homologene', 'medgen', 'mesh', 'nlmcatalog', 'omim', 'orgtrack', 'pmc', 'popset', 'proteinclusters', 'pcassay', 'protfam', 'pccompound', 'pcsubstance', 'seqannot', 'snp', 'sra', 'taxonomy', 'biocollections', 'gtr']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.einfo()\n",
    "record = Entrez.read(handle)\n",
    "record[\"DbList\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.1. PUBMED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PubMed bibliographic record'"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.einfo(db=\"pubmed\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"DbInfo\"][\"Description\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'36525710'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[\"DbInfo\"][\"Count\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['37810457', '37668712', '37577677', '36818783', '36245797', '36094101', '35497637', '35496474', '35402671', '34735950', '34484417', '34434786', '34189012', '33994075', '33902722', '33809815', '33242467', '32044951', '31762715', '31278684']"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"pubmed\", term=\"biopython\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"IdList\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Arora C', 'De Oliveira Rosa N', 'Matic M', 'Cascone M', 'Miglionico P', 'Raimondi F'] EXPANSION: a webserver to explore the functional consequences of protein-coding alternative splice variants in cancer genomics. 2023 Bioinformatics advances\n",
      "['Rustagi V', 'Gupta SRR', 'Bajaj M', 'Singh A', 'Singh IK'] PepAnalyzer: predicting peptide properties using its sequence. 2023 Oct Amino acids\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.esummary(db=\"pubmed\", id='37810457, 37668712')\n",
    "records = Entrez.parse(handle)\n",
    "\n",
    "\n",
    "for record in records:\n",
    "    print(record['AuthorList'], record['Title'], record['PubDate'], record['FullJournalName'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'<?xml version=\"1.0\" ?>\\n<!DOCTYPE PubmedArticleSet PUBLIC \"-//NLM//DTD PubMedArticle, 1st January 2023//EN\" \"https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_230101.dtd\">\\n<PubmedArticleSet>\\n<PubmedArticle><MedlineCitation Status=\"MEDLINE\" Owner=\"NLM\"><PMID Version=\"1\">19811691</PMID><DateCompleted><Year>2010</Year><Month>02</Month><Day>12</Day></DateCompleted><DateRevised><Year>2021</Year><Month>10</Month><Day>20</Day></DateRevised><Article PubModel=\"Electronic\"><Journal><ISSN IssnType=\"Electronic\">1471-2105</ISSN><JournalIssue CitedMedium=\"Internet\"><Volume>10 Suppl 11</Volume><Issue>Suppl 11</Issue><PubDate><Year>2009</Year><Month>Oct</Month><Day>08</Day></PubDate></JournalIssue><Title>BMC bioinformatics</Title><ISOAbbreviation>BMC Bioinformatics</ISOAbbreviation></Journal><ArticleTitle>Exploratory visual analysis of conserved domains on multiple sequence alignments.</ArticleTitle><Pagination><StartPage>S7</StartPage><MedlinePgn>S7</MedlinePgn></Pagination><ELocationID EIdType=\"doi\" ValidYN=\"Y\">10.1186/1471-2105-10-S11-S7</ELocationID><Abstract><AbstractText Label=\"BACKGROUND\" NlmCategory=\"BACKGROUND\">Multiple alignment of protein sequences can provide insight into sequence conservation across many species and thus allow identification of those sections of the sequence most critical to protein function. This insight can be augmented by joint display of conserved domains along the sequences. By fusing this metadata visually, biologists can analyze sequence conservation and functional motifs simultaneously and efficiently.</AbstractText><AbstractText Label=\"RESULTS\" NlmCategory=\"RESULTS\">We present MSAVis, a new approach combining luminance and hue for simultaneous visualization of conserved motifs and sequence alignment. Input for the algorithm is a multiple sequence alignment in a standard format. The NCBI Conserved Domain Database (CDD) is used for finding conserved domains along the alignment. The visualization quickly identifies conserved domains, and allows both macro (sequence-long) and micro (small amino-acid neighborhood) views.</AbstractText><AbstractText Label=\"CONCLUSION\" NlmCategory=\"CONCLUSIONS\">MSAVis utilizes two visual cues, luminance and hue, to facilitate at-a-glance summary of the conservation of a user-provided protein alignment while enabling multiple comparisons among functional domains. These visual cues are preattentive and separable so that the relationship between conservation strength and domain membership can be understood. The MSAVis software, written in Python and using BioPython and OpenGL, can be found at http://agbase.msstate.edu/tools/MSAVis.html.</AbstractText></Abstract><AuthorList CompleteYN=\"Y\"><Author ValidYN=\"Y\"><LastName>Jankun-Kelly</LastName><ForeName>T J</ForeName><Initials>TJ</Initials><AffiliationInfo><Affiliation>Institute for Digital Biology and Department of Computer Science and Engineering, Bagley College of Engineering, Mississippi State University, Mississippi, USA. tjk@acm.org</Affiliation></AffiliationInfo></Author><Author ValidYN=\"Y\"><LastName>Lindeman</LastName><ForeName>Andrew D</ForeName><Initials>AD</Initials></Author><Author ValidYN=\"Y\"><LastName>Bridges</LastName><ForeName>Susan M</ForeName><Initials>SM</Initials></Author></AuthorList><Language>eng</Language><PublicationTypeList><PublicationType UI=\"D016428\">Journal Article</PublicationType><PublicationType UI=\"D013486\">Research Support, U.S. Gov\\'t, Non-P.H.S.</PublicationType></PublicationTypeList><ArticleDate DateType=\"Electronic\"><Year>2009</Year><Month>10</Month><Day>08</Day></ArticleDate></Article><MedlineJournalInfo><Country>England</Country><MedlineTA>BMC Bioinformatics</MedlineTA><NlmUniqueID>100965194</NlmUniqueID><ISSNLinking>1471-2105</ISSNLinking></MedlineJournalInfo><ChemicalList><Chemical><RegistryNumber>0</RegistryNumber><NameOfSubstance UI=\"D011506\">Proteins</NameOfSubstance></Chemical></ChemicalList><CitationSubset>IM</CitationSubset><MeshHeadingList><MeshHeading><DescriptorName UI=\"D000595\" MajorTopicYN=\"N\">Amino Acid Sequence</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D019295\" MajorTopicYN=\"N\">Computational Biology</DescriptorName><QualifierName UI=\"Q000379\" MajorTopicYN=\"Y\">methods</QualifierName></MeshHeading><MeshHeading><DescriptorName UI=\"D030562\" MajorTopicYN=\"N\">Databases, Protein</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D008969\" MajorTopicYN=\"N\">Molecular Sequence Data</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D017434\" MajorTopicYN=\"Y\">Protein Structure, Tertiary</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D011506\" MajorTopicYN=\"N\">Proteins</DescriptorName><QualifierName UI=\"Q000737\" MajorTopicYN=\"N\">chemistry</QualifierName></MeshHeading><MeshHeading><DescriptorName UI=\"D016415\" MajorTopicYN=\"Y\">Sequence Alignment</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D020539\" MajorTopicYN=\"N\">Sequence Analysis, Protein</DescriptorName><QualifierName UI=\"Q000379\" MajorTopicYN=\"Y\">methods</QualifierName></MeshHeading></MeshHeadingList></MedlineCitation><PubmedData><History><PubMedPubDate PubStatus=\"entrez\"><Year>2009</Year><Month>10</Month><Day>9</Day><Hour>6</Hour><Minute>0</Minute></PubMedPubDate><PubMedPubDate PubStatus=\"pubmed\"><Year>2009</Year><Month>10</Month><Day>14</Day><Hour>6</Hour><Minute>0</Minute></PubMedPubDate><PubMedPubDate PubStatus=\"medline\"><Year>2010</Year><Month>2</Month><Day>13</Day><Hour>6</Hour><Minute>0</Minute></PubMedPubDate></History><PublicationStatus>epublish</PublicationStatus><ArticleIdList><ArticleId IdType=\"pubmed\">19811691</ArticleId><ArticleId IdType=\"pmc\">PMC3226196</ArticleId><ArticleId IdType=\"doi\">10.1186/1471-2105-10-S11-S7</ArticleId><ArticleId IdType=\"pii\">1471-2105-10-S11-S7</ArticleId></ArticleIdList><ReferenceList><Reference><Citation>Larkin M, Blackshields G, Brown N, Chenna R, McGettigan P, McWilliam H, Valentin F, Wallace I, Wilm A, Lopez R, Thompson J, Gibson T, Higgins D. ClustalW and ClustalX version 2. Bioinformatics. 2007;23(21):2947&#x2013;2948. doi: 10.1093/bioinformatics/btm404.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btm404</ArticleId><ArticleId IdType=\"pubmed\">17846036</ArticleId></ArticleIdList></Reference><Reference><Citation>A Conserved Domain Database and Search Service. http://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml</Citation></Reference><Reference><Citation>Marchler-Bauer A, Anderson J, Cherukuri P, DeWeese-Scott C, Geer L, Gwadz M, He S, Hurwitz D, Jackson J, Ke Z, Lanczycki C, Liebert C, Liu C, Lu F, Marchler G, Mullokandov M, Shoemaker B, Simonyan V, Song J, Thiessen P, Yamashita R, Yin J, Zhang D, Bryant S. CDD: A Conserved Domain Database for protein classification. Nucleic Acids Research. 2005. pp. D192&#x2013;D196.</Citation><ArticleIdList><ArticleId IdType=\"pmc\">PMC540023</ArticleId><ArticleId IdType=\"pubmed\">15608175</ArticleId></ArticleIdList></Reference><Reference><Citation>Lungu M, Xu K.  In: Human-Centered Visualization Environments, Volume 4417 of Lecture Notes in Computer Science. Kerren A, Ebert A, Meyer J, editor. Springer; 2006. Biomedical Information Visualization; pp. 311&#x2013;342.http://dx.doi.org/10.1007/978-3-540-71949-6_8</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1007/978-3-540-71949-6_8</ArticleId></ArticleIdList></Reference><Reference><Citation>Schneider TD, Stephens RM. Sequence Logos: A New Way to Display Consensus Sequences. Nucleic Acids Research. 1990;18:6097&#x2013;6100. doi: 10.1093/nar/18.20.6097.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/nar/18.20.6097</ArticleId><ArticleId IdType=\"pmc\">PMC332411</ArticleId><ArticleId IdType=\"pubmed\">2172928</ArticleId></ArticleIdList></Reference><Reference><Citation>Slack J, Hildebrand K, Munzner T, John KS. In: Proceedings of the German Conference on Bioinformatics. Giegerich R, Stoye J, editor. Vol. 53. 2004. SequenceJuxtaposer: Fluid Navigation For Large-Scale Sequence Comparison in Context; pp. 37&#x2013;42.</Citation></Reference><Reference><Citation>Mayor C, Brudno M, Schwartz JR, Poliakov A, Rubin EM, Frazer KA, Pachter LS, Dubchak I. VISTA: Visualizing Global DNA Sequence Alignments of Arbitrary Length. Bioinformatics. 2000;16(11):1046&#x2013;1047. doi: 10.1093/bioinformatics/16.11.1046.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/16.11.1046</ArticleId><ArticleId IdType=\"pubmed\">11159318</ArticleId></ArticleIdList></Reference><Reference><Citation>Shah N, Couronne O, Pennacchio LA, Brudno M, Batzoglou S, Bethel EW, Rubin EM, Hamann B, Dubchak I. Phylo-VISTA: Interactive visualization of multiple DNA sequence alignments. Bioinformatics. 2004;20(5):636&#x2013;643. doi: 10.1093/bioinformatics/btg459.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btg459</ArticleId><ArticleId IdType=\"pubmed\">15033870</ArticleId></ArticleIdList></Reference><Reference><Citation>Spell R, Brady R, Dietrich F. BARD: A visualization tool for biological sequence analysis. IEEE Symposium on Information Visualization 2003. 2003. pp. 219&#x2013;225.</Citation></Reference><Reference><Citation>InterPro. http://www.ebi.ac.uk/interpro/</Citation></Reference><Reference><Citation>Hunter S, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Bork P, Das U, Daugherty L, Duquenne L, Finn RD, Gough J, Haft D, Hulo N, Kahn D, Kelly E, Laugraud A, Letunic I, Lonsdale D, Lopez R, Madera M, Maslen J, McAnulla C, McDowall J, Mistry J, Mitchell A, Mulder N, Natale D, Orengo C, Quinn AF, Selengut JD, Sigrist CJA, Thimma M, Thomas PD, Valentin F, Wilson D, Wu CH, Yeats C. InterPro: The integrative protein signature database. Nucleic Acids Research. 2009. pp. D211&#x2013;D215.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/nar/gkn785</ArticleId><ArticleId IdType=\"pmc\">PMC2686546</ArticleId><ArticleId IdType=\"pubmed\">18940856</ArticleId></ArticleIdList></Reference><Reference><Citation>Consortium TU. The Universal Protein Resource (UniProt) Nucleic Acids Research. 2008. pp. D190&#x2013;D195.</Citation><ArticleIdList><ArticleId IdType=\"pmc\">PMC2238893</ArticleId><ArticleId IdType=\"pubmed\">18045787</ArticleId></ArticleIdList></Reference><Reference><Citation>UniProt. http://www.uniprot.org/</Citation></Reference><Reference><Citation>Clamp M, Cuff J, Searle S, Barton G. The Jalview Java alignment editor. Bioinformatics. 2004;20(3):426&#x2013;427. doi: 10.1093/bioinformatics/btg430.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btg430</ArticleId><ArticleId IdType=\"pubmed\">14960472</ArticleId></ArticleIdList></Reference><Reference><Citation>Felsensein J. PHYLIP. http://evolution.genetics.washington.edu/phylip.html</Citation></Reference><Reference><Citation>Henikoff S, Henikoff JG. Amino acid substitution matrices from protein blocks. Proc Natl Acad Sci USA. 1992;89(22):10915&#x2013;10919. doi: 10.1073/pnas.89.22.10915.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1073/pnas.89.22.10915</ArticleId><ArticleId IdType=\"pmc\">PMC50453</ArticleId><ArticleId IdType=\"pubmed\">1438297</ArticleId></ArticleIdList></Reference><Reference><Citation>McLaren K. The development of the CIE 1976 (L*a*b*) uniform colour-space and colour-difference formula. Journal of the Society of Dyers and Colourists. 1976;92:338&#x2013;341.</Citation></Reference><Reference><Citation>Ware C. Information Visualization: Perception for Design. 2. Morgan Kaufmann; 2004.</Citation></Reference><Reference><Citation>Pirolli P, Card SK. Information Foraging. Psychological Review. 1999;4:643&#x2013;674. doi: 10.1037/0033-295X.106.4.643.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1037/0033-295X.106.4.643</ArticleId></ArticleIdList></Reference><Reference><Citation>Cock P, Antao T, Chang J, Chapman B, Cox C, Dalke A, Friedberg I, Hamelryck T, Kauff F, Wilczynski B, de Hoon M. Biopython: Freely available Python tools for computational molecular biology and bioinformatics. Bioinformatics. 2009;25(11):1422&#x2013;1423. doi: 10.1093/bioinformatics/btp163.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btp163</ArticleId><ArticleId IdType=\"pmc\">PMC2682512</ArticleId><ArticleId IdType=\"pubmed\">19304878</ArticleId></ArticleIdList></Reference><Reference><Citation>PyOpenGL. http://pyopengl.sourceforge.net/</Citation></Reference><Reference><Citation>wxPython. http://wxpython.org/</Citation></Reference><Reference><Citation>McCarthy F, Wang N, Magee G, Nanduri B, Lawrence M, Camon E, Burrell D, Hill D, Dolan M, Williams W, Luthe D, Bridges S, Burgess S. AgBase: A Functional Genomics Resource for Agriculture. BMC Genomics. 2006;7(229)</Citation><ArticleIdList><ArticleId IdType=\"pmc\">PMC1618847</ArticleId><ArticleId IdType=\"pubmed\">16961921</ArticleId></ArticleIdList></Reference><Reference><Citation>MSAVis. http://agbase.msstate.edu/tools/MSAVis.html</Citation></Reference><Reference><Citation>Bailey TL, Boden M, Buske FA, Frith M, Grant CE, Clementi L, Ren J, Li WW, Noble WS. MEME SUITE: Tools for motif discovery and searching. Nucleic Acids Research. 2009. http://dx.doi.org/10.1093/nar/gkp335</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/nar/gkp335</ArticleId><ArticleId IdType=\"pmc\">PMC2703892</ArticleId><ArticleId IdType=\"pubmed\">19458158</ArticleId></ArticleIdList></Reference><Reference><Citation>The MEME Suite. http://meme.nbcr.net/</Citation></Reference><Reference><Citation>Nicholas KB, Nicholas HB, Deerfield DW. GeneDoc: Analysis and visualization of genetic variation. EMBNEW NEWS. 1997;4:14.</Citation></Reference><Reference><Citation>GeneDoc. http://www.nrbsc.org/gfx/genedoc/</Citation></Reference><Reference><Citation>Hall T. BioEdit Sequence Alignment Editor. http://www.mbio.ncsu.edu/BioEdit/bioedit.html</Citation></Reference></ReferenceList></PubmedData></PubmedArticle></PubmedArticleSet>'\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.efetch(db=\"pubmed\", id=\"19811691\")\n",
    "print(handle.read())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.2. Nucleotide"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['2623675418', '2623675407', '2623675395', '2623675383', '2623675371', '2623675359', '2623675347', '2623675335', '2623675323', '2623675311']"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"nucleotide\", retmax=10, term=\"Severe acute respiratory syndrome\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"IdList\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       OR840572               29648 bp    RNA     linear   VRL 22-NOV-2023\n",
      "DEFINITION  Severe acute respiratory syndrome coronavirus 2 isolate\n",
      "            SARS-CoV-2/human/USA/OSPHL09231/2023 ORF1ab polyprotein (ORF1ab),\n",
      "            ORF1a polyprotein (ORF1ab), surface glycoprotein (S), ORF3a protein\n",
      "            (ORF3a), envelope protein (E), membrane glycoprotein (M), ORF6\n",
      "            protein (ORF6), ORF7a protein (ORF7a), and ORF7b (ORF7b) genes,\n",
      "            complete cds; ORF8 gene, complete sequence; and nucleocapsid\n",
      "            phosphoprotein (N) and ORF10 protein (ORF10) genes, complete cds.\n",
      "ACCESSION   OR840572\n",
      "VERSION     OR840572.1\n",
      "KEYWORDS    .\n",
      "SOURCE      Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)\n",
      "  ORGANISM  Severe acute respiratory syndrome coronavirus 2\n",
      "            Viruses; Riboviria; Orthornavirae; Pisuviricota; Pisoniviricetes;\n",
      "            Nidovirales; Cornidovirineae; Coronaviridae; Orthocoronavirinae;\n",
      "            Betacoronavirus; Sarbecovirus; Severe acute respiratory\n",
      "            syndrome-related coronavirus.\n",
      "REFERENCE   1  (bases 1 to 29648)\n",
      "  AUTHORS   Tsaknaridis,L., Makris,V., Boyd,L., Weitemier,K., Geniza,M.,\n",
      "            Ojebode,A. and Evans,R.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (22-NOV-2023) Virology and Immunology, Oregon State\n",
      "            Public Health Laboratory, 7202 NE Evergreen Parkway, Suite 100,\n",
      "            Hillsboro, OR 97124, USA\n",
      "COMMENT     ##Assembly-Data-START##\n",
      "            Assembly Method       :: Medaka via Artic v. v1.2.1\n",
      "            Sequencing Technology :: Clear Labs_ONT\n",
      "            ##Assembly-Data-END##\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..29648\n",
      "                     /organism=\"Severe acute respiratory syndrome coronavirus\n",
      "                     2\"\n",
      "                     /mol_type=\"genomic RNA\"\n",
      "                     /isolate=\"SARS-CoV-2/human/USA/OSPHL09231/2023\"\n",
      "                     /host=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:2697049\"\n",
      "                     /country=\"USA: North America, Oregon,Unknown County\"\n",
      "                     /collection_date=\"2023-11-11\"\n",
      "     gene            212..21492\n",
      "                     /gene=\"ORF1ab\"\n",
      "     CDS             join(212..13405,13405..21492)\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /ribosomal_slippage\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF1ab polyprotein\"\n",
      "                     /protein_id=\"WPK49443.1\"\n",
      "                     /translation=\"MESLVPGFNEKTHVQLSLPVLQVRDVLVRGFGDSVEEVLSEARQ\n",
      "                     HLRDGTCGLVEVEKGVLPQLEQPYVFIKRSDARTAPHGHVMVELVAELEGIQYGRSGE\n",
      "                     TLGVLVPHVGEIPVAYRKVLLRKNGNKGAGGHXYGADLKSFDLGDELGTDPYEDFQEN\n",
      "                     WNTKHSSGVTRELMRELNGGAYTRYVDNNFCGPDGYPLECIKDLLARAGKASCTLSEQ\n",
      "                     LDFIDTKRGVYCCREHEHEIAWYTERSEKSYELQTPFEIKLAKKFDTFNGECPNFVFP\n",
      "                     LNSIIKTIQPRVEKKKLDGFMGRIRSVYPVASPNECNQMCLSTLMKCDHCGETSWQTG\n",
      "                     DFVKATCEFCGTENLTKEGATTCGYLPQNAVVKIYCPACHNSEVGPEHSLAEYHNESG\n",
      "                     LKTILRKGGRTIAFGGCVFSYVGCHNKCAYWVPRASANIGCNHTGVVGEGSEGLNDNL\n",
      "                     LEILQKEKVNINIVGDFKLNEEIAIILASFSASTSAFVETVKGLDYKAFKQIVESCGN\n",
      "                     FKVTKGKAKKSAWNIGEQKSILSPLYAFASEAARVVRSIFSRTLETAQNSVRVLQKAA\n",
      "                     ITILDGISQYSLRLIDAMMFTSDLATNNLVVMAYITGGVVQLTSQWLTNIFGTVYEKL\n",
      "                     KPVLDWLEEKFKEGVEFLRDGWEIVKFISTCACEIVGGQIVTCAKEIKESVQTFFKLV\n",
      "                     NKFLALCADSIIIGGAKLKALNLGETFVTHSKGLYRKCVKSREETGLLMPLKAPKEII\n",
      "                     FLEGETLPTEVLTEEVVLKTGDLQPLEQPTSEAVEAPLVGTPVCINGLMLLEIKDTEK\n",
      "                     YCALAPNMMVTNNTFTLKGGAPTKVTFGDDTVIEVQGYKSVNIIFELDERIDKVLNEK\n",
      "                     CSAYTVELGTEVNEFACVVADAVIKTLQPVSELLTPLGIDLDEWSMATYYLFDESGEF\n",
      "                     KLASHMYCSFYPPDEDEEEGDCEEEEFEPLTQYEYGTEDDYQGKPLEFGATSAALQPE\n",
      "                     EEQEEDWLDDDSQQTVGQQDGSEDNQTTTIQTIVEVQPQLEMELTPVVQTIEVNSFSG\n",
      "                     YLKLTDNVYIKNADIVEEAKKVKPTVVVNAANVYLKHGGGVAGALNKATNNAMQVESD\n",
      "                     DYIATNGPLKVGDSCVLSGHNLAKHCLHVVGPNVNKGEDIQLLKSAYENFNQHEVLLA\n",
      "                     PLLSAGIFGADSIHSLRVCVDTVRTNVYLAVFDKNLYDKLVSSFLEMKSEKQVEQKIA\n",
      "                     EIPKEEVKPFITESKPSVEQRKQDDKKIKACVEEVTTTLEETKFLTENLLLYIDINGN\n",
      "                     LHPDSATLVSDIDITFLKKDAPYIVGDVVQEGVLTAVVIPTKKASGTTEMLAKALRKV\n",
      "                     PTDNYITTYPGQGLNGYTVEEAKTVLKKCKSAFYILPSIISNEKQEILGTVSWNLREM\n",
      "                     LAHAEETRKLMPVCVETKAIVSTIQRKYKGIKIQEGVVDYGARFYFYTSKTTVASLIN\n",
      "                     TLNDLNETLVTMPLGYVTHGLNLEEAARYMRSLKVPATVSVSSPDAVTAYNGYLTSSS\n",
      "                     KTPEEHFIETISLAGSYKDWSYSGQSTQLGIEFLKRGDKSVYYTSNPTTFHLDGEVIT\n",
      "                     FDNLKTLLSLREVRTIKVFTTVDNINLHTQVVDMSMTYGQQFGPTYLDGADVTKIKPH\n",
      "                     NSHEGKTFYVLPNDDTLRVEAFEYYHTTDPSFLGRYMSALNHTKKWKYPQVNGLTSIK\n",
      "                     WADNNCYLATALLTLQQIELKFNPPALQDAYYRARAGEAANFCALILAYCNKTVGELG\n",
      "                     DVRETMSYLFQHANLDSCKRVLNVVCKTCGQQQTTLKGVEAVMYMGTLSYEQFKKGVQ\n",
      "                     IPCTCGKQATKYLVQQESPFVMMSAPPAQYELKHGTFTCASEYTGNYQCGHYKHITSK\n",
      "                     ETLYCIDGALLTKSSEYKGPITDVFYKENSYTTTIKPVTYKLDGVVCTEIDPKLDNYY\n",
      "                     KKDNSYFTEQPIDLVPNQPYPNASFDNFKFVCDNIKFADDLNQLTGYKKPASRELKVT\n",
      "                     FFPDLNGDVVAIDYKHYTPSFKKGAKLLHKPIVWHVNNATNKATYKPNTWCIRCLWST\n",
      "                     KPVETSNSFDVLKSEDAQGMDNLACEDLKPVSEEVVENPTIQKDVLECNVKTTEVVGD\n",
      "                     IILKPANNSLKITEEVGHTDLMAAYVDNSSLTIKKPNELSRVLGLKTLATHGLAAVNS\n",
      "                     VPWDTIANYAKPFLNKVVSTTTNIVTRCLNRVCTNYMPYFFTLLLQLCTFTRSTNSRI\n",
      "                     KASMPTTIAKNTVKSVGKFCLEASFNYLKSPNFSKLINIIIWFLLLSVCLGSLIYSTA\n",
      "                     ALGVLMSNLGMPSYCTGYREGYLNSTNVTIATYCTGSIPCSVCLSGLDSLDTYPSLET\n",
      "                     IQITISSFKWDLTAFGLVAEWFLAYILFTRFFYVLGLAAIMQLFFSYFAVHFISNSWL\n",
      "                     MWLIINLVQMAPISAMVRMYIFFASFYYVWKSYVHVVDGCNSSTCMMCYKRNRATRVE\n",
      "                     CTTIVNGVRRSFYVYANGGKGFCKLHNWNCVNCDTFCAGSTFISDEVARDLSLQFKRP\n",
      "                     INPTDQSSYIVDSVTVKNGSIHLYFDKAGQKTYERHSLSHFVNLDNLRANNTKGSLPI\n",
      "                     NVIVFDGKSKCEESSAKSASVYYSQLMCQPILLLDQALVSDVGDSAEVAVKMFDAYVN\n",
      "                     TFSSTFNVPMEKLKTLVATAEAELAKNVSLDNVLSTFISAARQGFVDSDVETKDVVEC\n",
      "                     LKLSHQSDIEVTGDSCNNYMLTYNKVENMTPRDLGACIDCSARHINAQVAKSHNIALI\n",
      "                     WNVKDFMSLSEQLRKQIRSAAKKNNLPFKLTCATTRQVVNVVTTKIALKGGKIVNNWL\n",
      "                     KQLIKVTLVFLFVAAIFYLITPVHVMSKHTDFSSEIIGYKAIDGGVTRDIASTDTCFA\n",
      "                     NKHADFDTWFSQRGGSYTNDKACPLIAAVITREVGFVVPGLPGTILRTTNGDFLHFLP\n",
      "                     RVFSAVGNICYTPSKLIEYTDFATSACVLAAECTIFKDASGKPVPYCYDTNVLEGSVA\n",
      "                     YESLRPDTRYVLMDGSIIQFPNTYLEGSVRVVTTFDSEYCRHGTCERSEAGVCVSTSG\n",
      "                     RWVLNNDYYRSLPGVFCGVDAVNLFTNMFTPLIQPIGALDISASIVAGGIVAIVVTCL\n",
      "                     AYYFMRFRRAFGEYSHVVAFNTLLFLMSFIVLCLTPVYSFLPGVYSVIYLYLTFYLTN\n",
      "                     DVSFLAHIQWMVMFTPLVPFWITIAYIICISTKHFYWFFSNYLKRRVVFNGVSFSTFE\n",
      "                     EAALCTFLLNKEMYLKLRSDVLLPFTQYNRYLALYNKYKYFSGAMDTTSYREAACCHL\n",
      "                     AKALNDFSNSGSDVLYQXXXXSITSAVLQSGFRKMAFPSGKVEXXXXXXXXXXXXXXX\n",
      "                     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n",
      "                     XXXXXXXXXXXXXXXXXXXQPGQTFSVLACYNGSPSGVYQCAMRHNFTIKGSFLNGSC\n",
      "                     GSVGFNIDYDCVSFCYMHHMELPTGVHAGTDLEGNFYGPFVDRQTAQAAGTDTTITVN\n",
      "                     VLAWLYAAVINGDRWFLNRFTTTLNDFNLVAMKYNYEPLTQDHVDILGPLSAQTGIAV\n",
      "                     LDMCASLKELLQNGMNGRTILGSALLEDEFTPFDVVRQCSGVTFQSAVKRTIKGTHHW\n",
      "                     LLLTILTSLLVLVQSTQWSLFFFLYENAFLPFAMGIIAMSAFAMMFVKHKHAFLCLFL\n",
      "                     LPSLATVAYFNMVYMPASWVMRIMTWLDMVDTSLKLKDCVMYASAVVLLILMTARTVY\n",
      "                     DDGARRVWTLMNVLTLVYKVYYGNALDQAISMWALIISVTSNYSGVVTTVMFLARGIV\n",
      "                     FMCVEYCPIFFITGNTLQCIMLVYCFLGYFCTCYFGLFCLLNRYFRLTLGVYDYLVST\n",
      "                     QEFRYMNSQGLFPPKNSIDAFKLNIKLLGVGGKPCIKVATVQSKMSDVKCTSVVLLSV\n",
      "                     LQQLRVESSSKLWAQCVQLHNDILLAKDTTEAFEKMVSLLSVLLSMQGAVDINKLCEE\n",
      "                     MLDNRATLQAIASEFSSLPSYAAFATAQEAYEQAVANGDSEVVLKKLKKSLNVAKSEF\n",
      "                     DRDAAMQRKLEKMADQAMTQMYKQARSEDKRAKVTSAMQTMLFTMLRKLDNDALNNII\n",
      "                     NNARDGCVPLNIIPLTTAAKLMVVIPDYNTYKNTCDGTTFTYASALWEIQQVVDADSK\n",
      "                     IVQLSEISMDNSPNLAWPLIVTALRANSAVKLQNNELSPVALRQMSCAAGTTQTACTD\n",
      "                     DNALAYYNTTKGGRFVLALLSDLQDLKWARFPKSDGTGTIYTELEPPCRFVTDTPKGP\n",
      "                     KVKYLYFIKGLNNLNRGMVLGSLAATVRLQAGNATEVPANSTVLSFCAFAVDAAKAYK\n",
      "                     DYLASGGQPITNCVKMLCTHTGTGQAITVTPEANMDQESFGGASCCLYCRCHIDHPNP\n",
      "                     KGFCDLKGKYVQIPTTCANDPVGFTLKNTVCTVCGMWKGYGCSCDQLREPMLQSADAQ\n",
      "                     SFLNRVCGVSAARLTPCGTGTSTDVVYRAFDIYNDKVAGFAKFLKTNCCRFQEKDEDD\n",
      "                     NLIDSYFVVKRHTFSNYQHEETIYNLLKDCLAVAKHDFFKFRIDGDMVPHISRQRLTK\n",
      "                     YTMADLVYALRHFDEGNCDTLKEILVTYNCCDDDYFNKKDWYDFVENPDILRVYANLG\n",
      "                     ERVRQALLKTVQFCDAMRNAGIVGVLTLDNQDLNGNWYDFGDFIQTTPGSGVPVVDSY\n",
      "                     YSLLMPILTLTRALTAESHVDTDLTKPYIKWDLLKYDFTEERLKLFDRYFKYWDQTYH\n",
      "                     PNCVNCLDDRCILHCANFNVLFSTVFPLTSFGPLVRKIFVDGVPFVVSTGYHFRELGV\n",
      "                     VHNQDVNLHSSRLSFKELLVYAADPAMHAASGNLLLDKRTTCFSVAALTNNVAFQTVK\n",
      "                     PGNFNKDFYDFAVSKGFFKEGSSVELKHFFFAQDGNAAISDYDYYRYNLPTMCDIRQL\n",
      "                     LFVVEVVDKYFDCYDGGCINANQVIVNNLDKSAGFPFNKWGKARLYYDSMSYEDQDAL\n",
      "                     FAYTKRNVIPTITQMNLKYAISAKNRARTVAGVSICSTMTNRQFHQKLLKSIAATRGA\n",
      "                     TVVIGTSKFYGGWHNMLKTVYSDVENPHLMGWDYPKCDRAMPNMLRIMASLVLARKHT\n",
      "                     TCCSLSHRFYRLANECAQVLSEMVMCGSSLYVKPGGTSSGDATTAYANSVFNICQAVT\n",
      "                     ANVNALLSTDGNKIADKYVRNLQHRLYECLYRNRDVDTDFVNEFYAYLRKHFSMMILS\n",
      "                     DDAVVCFNSTYASQGLVASIKNFKSVLYYQNNVFMSEAKCWTETDLTKGPHEFCSQHT\n",
      "                     MLVKQGDDYVYLPYPDPSRILGAGCFVDDIVKTDGTLMIERFVSLAIDAYPLTKHPNQ\n",
      "                     EYADVFHLYLQYIRKLHDELTGHMLDMYSVMLTNDNTSRYWEPEFYEAMYTPHTVLQA\n",
      "                     VGACVLCNSQTSLRCGACIRRPFLCCKCCYDHVIPTSHKLVLSVNPYVCNAPGCDVTD\n",
      "                     VTQLYLGGMSYYCKSHKPPISFPLCANGQVFGLYKNTCVGSDNVTDFNAIATCDWTNA\n",
      "                     GDYILANTCTERLKLFAAETLKATEETFKLSYGIATVREVLSDRELHLSWEVGKPRPP\n",
      "                     LNRNYVFTGYRVTKNSKVQIGEYTFEKGDYGDAVVYRGTTTYKLNVGDYFVLTSHTVM\n",
      "                     PLSAPTLVPQEHYVRITGLYPTLNISDEFSSNVANYQKVGMQKYSTLQGPPGTGKSHF\n",
      "                     AIGLALYYPSARIVYTACSHAAVDALCEKALKYLPIDKCSRIIPARARVECFDKFKVN\n",
      "                     STLEQYVFCTVNALPETTADIVVFDEISMATNYDLSVVNARLCAKHYVYIGDPAQLPA\n",
      "                     PRTLLTKGTLEPEYFNSVCRLMKTIGPDMFLGTCRRCPAEIVDTVSALVYDNKLKAHK\n",
      "                     DKSAQCFKMFYKGVITHDVSSAINRPQIGVVREFLTRNPAWRKAVFISPYNSQNAVAS\n",
      "                     KILGLPTQTVDSSQGSEYDYVIFTQTTETAHSCNVNRFNVAITRAKVGILCIMSDRDL\n",
      "                     YDKLQFTSLEIPRRNVATLQAENVTGLFKDCSKVITGLHPTQAPTHLSVDTKFKTEGL\n",
      "                     CVDVPGIPKDMTYRRLISMMGFKMNYQVNGYPNMFITREEAIRHVRAWIGFDVEGCHA\n",
      "                     TREAVGTNLPLQLGFSTGVNLVAVPTGYVDTPNNTDFSRVSAKPPPGDQFKHLIPLMY\n",
      "                     KGLPWNVVRIKIVQMLSDTLKNLSDRVVFVLWAHGFELTSMKYFVKIGPERTCCLCDR\n",
      "                     RATCFSTASYTYACWHHSIGFDYVYNPFMIDVQQWGFTGNLQSNHDLYCQVHGNAHVA\n",
      "                     SCDAIMTRCLAVHECFVKRVDWTIEYPIIGDELKINAACRKVQHMVVKAALLADKFPV\n",
      "                     LHDIGNPKAIKCVPQADVEWKFYDAQPCSDKAYKIEELFYSYATHSDKFTDGVCLFWN\n",
      "                     CNVDRYPANSIVCRFDTRVLSNLNLPGCDGGSLYVNKHAFHTPAFDKSAFVNLKQLPF\n",
      "                     FYYSDSPCESHGKQVVSDIDYVPLKSATCITRCNLGGAVCRHHANEYRLYLDAYNMMI\n",
      "                     SAGFSLWVYKQFDTYNLWNTFTRLQSLENVAFNVVNKGHFDGQQGEVPVSIINNTVYT\n",
      "                     KVDGVDVELFENKTTLPVNVAFELWAKRNIKPVPEVKILNNLGVDIAANTVIWDYKRD\n",
      "                     APAHISTIGVCSMTDIAKKPIETICAPLTVFFDGRVDGQVDLFRNARNGVLITEGSVK\n",
      "                     GLQPSVGPKQASLNGVTLIGEAVKTQFNYYKKVDGVVQQLPETYFTQSRNLQEFKPRS\n",
      "                     QMEIDFLELAMDEFIERYKLEGYAFEHIVYGDFSHSQLGGLHLLIGLAKRFKESPFEL\n",
      "                     EDFIPMDSTVKNYFITDAQTGSSKCVCSVIDLLLDDFVEIIKSQDLSVVSKVVKVTID\n",
      "                     YTEISFMLWCKDGHVETFYPKLQSSQAWQPGVAMPNLYKMQRMLLEKCDLQNYGDSAT\n",
      "                     LPKGIMMNVAKYTQLCQYLNTLTLAVPYNMRVIHFGAGSDKGVAPGTAVLRQWLPTGT\n",
      "                     LLVDSDLNDFVSDADSTLIGDCATVHTANKWDLIISDMYDPKTKNVTKENDSKEGFFT\n",
      "                     YICGFIQQKLALGGSVAIKITEHSWNADLYKLMGHFAWWTAFVTNVNASSSEAFLIGC\n",
      "                     NYLGKPREQIDGYVMHANYIFWRNTNPIQLSSYSLFDMSKFPLKLRGTAVMSLKEGQI\n",
      "                     NDMILSLLSKGRLIIRENNRVVISSDVLVNN\"\n",
      "     mat_peptide     212..751\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"leader protein\"\n",
      "     mat_peptide     752..2665\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp2\"\n",
      "     mat_peptide     2666..8500\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp3\"\n",
      "     mat_peptide     8501..10000\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp4\"\n",
      "     mat_peptide     10001..10918\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"3C-like proteinase\"\n",
      "     mat_peptide     10919..11779\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp6\"\n",
      "     mat_peptide     11780..12028\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp7\"\n",
      "     mat_peptide     12029..12622\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp8\"\n",
      "     mat_peptide     12623..12961\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp9\"\n",
      "     mat_peptide     12962..13378\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp10\"\n",
      "     mat_peptide     join(13379..13405,13405..16173)\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"RNA-dependent RNA polymerase\"\n",
      "     mat_peptide     16174..17976\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"helicase\"\n",
      "     mat_peptide     17977..19557\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"3'-to-5' exonuclease\"\n",
      "     mat_peptide     19558..20595\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"endoRNAse\"\n",
      "     mat_peptide     20596..21489\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"2'-O-ribose methyltransferase\"\n",
      "     CDS             212..13420\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF1a polyprotein\"\n",
      "                     /protein_id=\"WPK49444.1\"\n",
      "                     /translation=\"MESLVPGFNEKTHVQLSLPVLQVRDVLVRGFGDSVEEVLSEARQ\n",
      "                     HLRDGTCGLVEVEKGVLPQLEQPYVFIKRSDARTAPHGHVMVELVAELEGIQYGRSGE\n",
      "                     TLGVLVPHVGEIPVAYRKVLLRKNGNKGAGGHXYGADLKSFDLGDELGTDPYEDFQEN\n",
      "                     WNTKHSSGVTRELMRELNGGAYTRYVDNNFCGPDGYPLECIKDLLARAGKASCTLSEQ\n",
      "                     LDFIDTKRGVYCCREHEHEIAWYTERSEKSYELQTPFEIKLAKKFDTFNGECPNFVFP\n",
      "                     LNSIIKTIQPRVEKKKLDGFMGRIRSVYPVASPNECNQMCLSTLMKCDHCGETSWQTG\n",
      "                     DFVKATCEFCGTENLTKEGATTCGYLPQNAVVKIYCPACHNSEVGPEHSLAEYHNESG\n",
      "                     LKTILRKGGRTIAFGGCVFSYVGCHNKCAYWVPRASANIGCNHTGVVGEGSEGLNDNL\n",
      "                     LEILQKEKVNINIVGDFKLNEEIAIILASFSASTSAFVETVKGLDYKAFKQIVESCGN\n",
      "                     FKVTKGKAKKSAWNIGEQKSILSPLYAFASEAARVVRSIFSRTLETAQNSVRVLQKAA\n",
      "                     ITILDGISQYSLRLIDAMMFTSDLATNNLVVMAYITGGVVQLTSQWLTNIFGTVYEKL\n",
      "                     KPVLDWLEEKFKEGVEFLRDGWEIVKFISTCACEIVGGQIVTCAKEIKESVQTFFKLV\n",
      "                     NKFLALCADSIIIGGAKLKALNLGETFVTHSKGLYRKCVKSREETGLLMPLKAPKEII\n",
      "                     FLEGETLPTEVLTEEVVLKTGDLQPLEQPTSEAVEAPLVGTPVCINGLMLLEIKDTEK\n",
      "                     YCALAPNMMVTNNTFTLKGGAPTKVTFGDDTVIEVQGYKSVNIIFELDERIDKVLNEK\n",
      "                     CSAYTVELGTEVNEFACVVADAVIKTLQPVSELLTPLGIDLDEWSMATYYLFDESGEF\n",
      "                     KLASHMYCSFYPPDEDEEEGDCEEEEFEPLTQYEYGTEDDYQGKPLEFGATSAALQPE\n",
      "                     EEQEEDWLDDDSQQTVGQQDGSEDNQTTTIQTIVEVQPQLEMELTPVVQTIEVNSFSG\n",
      "                     YLKLTDNVYIKNADIVEEAKKVKPTVVVNAANVYLKHGGGVAGALNKATNNAMQVESD\n",
      "                     DYIATNGPLKVGDSCVLSGHNLAKHCLHVVGPNVNKGEDIQLLKSAYENFNQHEVLLA\n",
      "                     PLLSAGIFGADSIHSLRVCVDTVRTNVYLAVFDKNLYDKLVSSFLEMKSEKQVEQKIA\n",
      "                     EIPKEEVKPFITESKPSVEQRKQDDKKIKACVEEVTTTLEETKFLTENLLLYIDINGN\n",
      "                     LHPDSATLVSDIDITFLKKDAPYIVGDVVQEGVLTAVVIPTKKASGTTEMLAKALRKV\n",
      "                     PTDNYITTYPGQGLNGYTVEEAKTVLKKCKSAFYILPSIISNEKQEILGTVSWNLREM\n",
      "                     LAHAEETRKLMPVCVETKAIVSTIQRKYKGIKIQEGVVDYGARFYFYTSKTTVASLIN\n",
      "                     TLNDLNETLVTMPLGYVTHGLNLEEAARYMRSLKVPATVSVSSPDAVTAYNGYLTSSS\n",
      "                     KTPEEHFIETISLAGSYKDWSYSGQSTQLGIEFLKRGDKSVYYTSNPTTFHLDGEVIT\n",
      "                     FDNLKTLLSLREVRTIKVFTTVDNINLHTQVVDMSMTYGQQFGPTYLDGADVTKIKPH\n",
      "                     NSHEGKTFYVLPNDDTLRVEAFEYYHTTDPSFLGRYMSALNHTKKWKYPQVNGLTSIK\n",
      "                     WADNNCYLATALLTLQQIELKFNPPALQDAYYRARAGEAANFCALILAYCNKTVGELG\n",
      "                     DVRETMSYLFQHANLDSCKRVLNVVCKTCGQQQTTLKGVEAVMYMGTLSYEQFKKGVQ\n",
      "                     IPCTCGKQATKYLVQQESPFVMMSAPPAQYELKHGTFTCASEYTGNYQCGHYKHITSK\n",
      "                     ETLYCIDGALLTKSSEYKGPITDVFYKENSYTTTIKPVTYKLDGVVCTEIDPKLDNYY\n",
      "                     KKDNSYFTEQPIDLVPNQPYPNASFDNFKFVCDNIKFADDLNQLTGYKKPASRELKVT\n",
      "                     FFPDLNGDVVAIDYKHYTPSFKKGAKLLHKPIVWHVNNATNKATYKPNTWCIRCLWST\n",
      "                     KPVETSNSFDVLKSEDAQGMDNLACEDLKPVSEEVVENPTIQKDVLECNVKTTEVVGD\n",
      "                     IILKPANNSLKITEEVGHTDLMAAYVDNSSLTIKKPNELSRVLGLKTLATHGLAAVNS\n",
      "                     VPWDTIANYAKPFLNKVVSTTTNIVTRCLNRVCTNYMPYFFTLLLQLCTFTRSTNSRI\n",
      "                     KASMPTTIAKNTVKSVGKFCLEASFNYLKSPNFSKLINIIIWFLLLSVCLGSLIYSTA\n",
      "                     ALGVLMSNLGMPSYCTGYREGYLNSTNVTIATYCTGSIPCSVCLSGLDSLDTYPSLET\n",
      "                     IQITISSFKWDLTAFGLVAEWFLAYILFTRFFYVLGLAAIMQLFFSYFAVHFISNSWL\n",
      "                     MWLIINLVQMAPISAMVRMYIFFASFYYVWKSYVHVVDGCNSSTCMMCYKRNRATRVE\n",
      "                     CTTIVNGVRRSFYVYANGGKGFCKLHNWNCVNCDTFCAGSTFISDEVARDLSLQFKRP\n",
      "                     INPTDQSSYIVDSVTVKNGSIHLYFDKAGQKTYERHSLSHFVNLDNLRANNTKGSLPI\n",
      "                     NVIVFDGKSKCEESSAKSASVYYSQLMCQPILLLDQALVSDVGDSAEVAVKMFDAYVN\n",
      "                     TFSSTFNVPMEKLKTLVATAEAELAKNVSLDNVLSTFISAARQGFVDSDVETKDVVEC\n",
      "                     LKLSHQSDIEVTGDSCNNYMLTYNKVENMTPRDLGACIDCSARHINAQVAKSHNIALI\n",
      "                     WNVKDFMSLSEQLRKQIRSAAKKNNLPFKLTCATTRQVVNVVTTKIALKGGKIVNNWL\n",
      "                     KQLIKVTLVFLFVAAIFYLITPVHVMSKHTDFSSEIIGYKAIDGGVTRDIASTDTCFA\n",
      "                     NKHADFDTWFSQRGGSYTNDKACPLIAAVITREVGFVVPGLPGTILRTTNGDFLHFLP\n",
      "                     RVFSAVGNICYTPSKLIEYTDFATSACVLAAECTIFKDASGKPVPYCYDTNVLEGSVA\n",
      "                     YESLRPDTRYVLMDGSIIQFPNTYLEGSVRVVTTFDSEYCRHGTCERSEAGVCVSTSG\n",
      "                     RWVLNNDYYRSLPGVFCGVDAVNLFTNMFTPLIQPIGALDISASIVAGGIVAIVVTCL\n",
      "                     AYYFMRFRRAFGEYSHVVAFNTLLFLMSFIVLCLTPVYSFLPGVYSVIYLYLTFYLTN\n",
      "                     DVSFLAHIQWMVMFTPLVPFWITIAYIICISTKHFYWFFSNYLKRRVVFNGVSFSTFE\n",
      "                     EAALCTFLLNKEMYLKLRSDVLLPFTQYNRYLALYNKYKYFSGAMDTTSYREAACCHL\n",
      "                     AKALNDFSNSGSDVLYQXXXXSITSAVLQSGFRKMAFPSGKVEXXXXXXXXXXXXXXX\n",
      "                     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n",
      "                     XXXXXXXXXXXXXXXXXXXQPGQTFSVLACYNGSPSGVYQCAMRHNFTIKGSFLNGSC\n",
      "                     GSVGFNIDYDCVSFCYMHHMELPTGVHAGTDLEGNFYGPFVDRQTAQAAGTDTTITVN\n",
      "                     VLAWLYAAVINGDRWFLNRFTTTLNDFNLVAMKYNYEPLTQDHVDILGPLSAQTGIAV\n",
      "                     LDMCASLKELLQNGMNGRTILGSALLEDEFTPFDVVRQCSGVTFQSAVKRTIKGTHHW\n",
      "                     LLLTILTSLLVLVQSTQWSLFFFLYENAFLPFAMGIIAMSAFAMMFVKHKHAFLCLFL\n",
      "                     LPSLATVAYFNMVYMPASWVMRIMTWLDMVDTSLKLKDCVMYASAVVLLILMTARTVY\n",
      "                     DDGARRVWTLMNVLTLVYKVYYGNALDQAISMWALIISVTSNYSGVVTTVMFLARGIV\n",
      "                     FMCVEYCPIFFITGNTLQCIMLVYCFLGYFCTCYFGLFCLLNRYFRLTLGVYDYLVST\n",
      "                     QEFRYMNSQGLFPPKNSIDAFKLNIKLLGVGGKPCIKVATVQSKMSDVKCTSVVLLSV\n",
      "                     LQQLRVESSSKLWAQCVQLHNDILLAKDTTEAFEKMVSLLSVLLSMQGAVDINKLCEE\n",
      "                     MLDNRATLQAIASEFSSLPSYAAFATAQEAYEQAVANGDSEVVLKKLKKSLNVAKSEF\n",
      "                     DRDAAMQRKLEKMADQAMTQMYKQARSEDKRAKVTSAMQTMLFTMLRKLDNDALNNII\n",
      "                     NNARDGCVPLNIIPLTTAAKLMVVIPDYNTYKNTCDGTTFTYASALWEIQQVVDADSK\n",
      "                     IVQLSEISMDNSPNLAWPLIVTALRANSAVKLQNNELSPVALRQMSCAAGTTQTACTD\n",
      "                     DNALAYYNTTKGGRFVLALLSDLQDLKWARFPKSDGTGTIYTELEPPCRFVTDTPKGP\n",
      "                     KVKYLYFIKGLNNLNRGMVLGSLAATVRLQAGNATEVPANSTVLSFCAFAVDAAKAYK\n",
      "                     DYLASGGQPITNCVKMLCTHTGTGQAITVTPEANMDQESFGGASCCLYCRCHIDHPNP\n",
      "                     KGFCDLKGKYVQIPTTCANDPVGFTLKNTVCTVCGMWKGYGCSCDQLREPMLQSADAQ\n",
      "                     SFLNGFAV\"\n",
      "     mat_peptide     212..751\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"leader protein\"\n",
      "     mat_peptide     752..2665\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp2\"\n",
      "     mat_peptide     2666..8500\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp3\"\n",
      "     mat_peptide     8501..10000\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp4\"\n",
      "     mat_peptide     10001..10918\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"3C-like proteinase\"\n",
      "     mat_peptide     10919..11779\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp6\"\n",
      "     mat_peptide     11780..12028\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp7\"\n",
      "     mat_peptide     12029..12622\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp8\"\n",
      "     mat_peptide     12623..12961\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp9\"\n",
      "     mat_peptide     12962..13378\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp10\"\n",
      "     mat_peptide     13379..13417\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp11\"\n",
      "     gap             10043..10316\n",
      "                     /estimated_length=274\n",
      "     stem_loop       13413..13440\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /note=\"Coronavirus frameshifting stimulation element\n",
      "                     stem-loop 1\"\n",
      "     stem_loop       13425..13479\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /note=\"Coronavirus frameshifting stimulation element\n",
      "                     stem-loop 2\"\n",
      "     gene            21500..25309\n",
      "                     /gene=\"S\"\n",
      "     CDS             21500..25309\n",
      "                     /gene=\"S\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"surface glycoprotein\"\n",
      "                     /protein_id=\"WPK49445.1\"\n",
      "                     /translation=\"MFVFLVLLPLVSSQCVNLITRTQSYTNSFTRGVYYPDKVFRSSV\n",
      "                     LHSTQDLFLPFFSNVTWFHAIHVSGTNGTKRFDNPALPFNDGVYFASTEKSNIIRGWI\n",
      "                     FGTTLDSKTQSLLIVNNATNVVIKVCEFQFCNDPFLDVYQKNNKSWMESEFRVYSSAN\n",
      "                     NCTFEYVSQPFLMDLVGKEGNFKNLREFVFKNIDGYFKIYSKHTPINLERDLPQGFSA\n",
      "                     LEPLVDLPIGINITRFQTLLALHRSYLTPVDSSSGWTAGAAAYYVGYLQPRTFLLKYN\n",
      "                     ENGTITDAVDCALDPLSETKCTLNSFTVEKGIYQTSNFRVQPTESIVRFPNITNLCPF\n",
      "                     HEVFNATTFASVYAWNRKRISNCVADYSVIYNFAPFFAFKCYGVSPTKLNDLCFTNVY\n",
      "                     ADSFVIRGNEVSQIAPGQTGNIADYNYKLPDDFTGCVIAWNSNKLDSKPSGNYNYLYR\n",
      "                     LFRKSKLKPFERDISTEIYQAGNRPCNGVAGPNCYSPLQSYGFRPTYGVGHQPYRVVV\n",
      "                     LSFELLHAPATVCGPKKSTNLVKNKCVNFNFNGLTGTGVLTESNKKFLPFQQFGRDIA\n",
      "                     DTTDAVRDPQTLEILDITPCSFGGVSVITPGTNTSNQVAVLYQGVNCTEVPVAIHADQ\n",
      "                     LTPTWRVYSTGSNVFQTRAGCLIGAEYVNNSYECDIPIGAGICASYQTQTKSHRRARS\n",
      "                     VASQSIIAYTMSLGAENSVAYSNNSIAIPTNFTISVTTEILPVSMTKTSVDCTMYICG\n",
      "                     DSTECSNLLLQYGSFCTQLKRALTGIAVEQDKNTQEVFAQVKQIYKTPPIKYFGGFNF\n",
      "                     SQILPDPSKPSKRSFIEDLLFNKVTLADAGFIKQYGDCLGDIAARDLICAQKFNGLTV\n",
      "                     LPPLLTDEMIAQYTSALLAGTITSGWTFGAGAALQIPFAMQMAYRFNGIGVTQNVLYE\n",
      "                     NQKLIANQFNSAIGKIQDSLSSTASALGKLQDVVNHNAQALNTLVKQLSSKFGAISSV\n",
      "                     LNDILSRLDKVEAEVQIDRLITGRLQSLQTYVTQQLIRAAEIRASANLAATKMSECVL\n",
      "                     GQSKRVDFCGKGYHLMSFPQSAPHGVVFLHVTYVPAQEKNFTTAPAICHDGKAHFPRE\n",
      "                     GVFVSNGTHWFVTQRNFYEPQIITTDNTFVSGNCDVVIGIVNNTVYDPLQPELDSFKE\n",
      "                     ELDKYFKNHTSPDVDLGDISGINASVVNIQKEIDRLNEVAKNLNESLIDLQELGKYEQ\n",
      "                     YIKWPWYIWLGFIAGLIAIVMVTIMLCCMTSCCSCLKGCCSCGSCCKFDEDDSEPVLK\n",
      "                     GVKLHYT\"\n",
      "     gene            25318..26145\n",
      "                     /gene=\"ORF3a\"\n",
      "     CDS             25318..26145\n",
      "                     /gene=\"ORF3a\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF3a protein\"\n",
      "                     /protein_id=\"WPK49446.1\"\n",
      "                     /translation=\"MDLFMRIFTIGTVTLKQGEIKDATPSDFVRATATIPIQASLPFG\n",
      "                     WLIVGVALLAVFQSASKIITLKKRWQLALSKGVHFVCNLLLLFVTVYSHLLLVAAGLE\n",
      "                     APFLYLYALVYFLQSINFVRIIMRLWLCWKCRSKNPLLYDANYFLCWHTNCYDYCIPY\n",
      "                     NSVTSSIVITSGDGTTSPISEHDYQIGGYTEKWESGVKDCVVLHSYFTSDYYQLYSTQ\n",
      "                     LSTDIGVEHVTFFIYNKIVDEPEEHVQIHTIDGSSGVVNPVMEPIYDEPTTTTSVPL\"\n",
      "     gene            26170..26397\n",
      "                     /gene=\"E\"\n",
      "     CDS             26170..26397\n",
      "                     /gene=\"E\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"envelope protein\"\n",
      "                     /protein_id=\"WPK49447.1\"\n",
      "                     /translation=\"MYSFVSEEIGALIVNSVLLFLAFVVFLLVTLAILTALRLCAYCC\n",
      "                     NIVNVSLVKPSFYVYSRVKNLNSSRVPDLLV\"\n",
      "     gene            26448..27116\n",
      "                     /gene=\"M\"\n",
      "     CDS             26448..27116\n",
      "                     /gene=\"M\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"membrane glycoprotein\"\n",
      "                     /protein_id=\"WPK49448.1\"\n",
      "                     /translation=\"MADSNGTITVEELKKLLEEWNLVIGFLFLTWICLLQFAYANRNR\n",
      "                     FLYIIKLIFLWLLWPVTLTCFVLAAVYRINWITGGIAIAMACLVGLMWLSYFIASFRL\n",
      "                     FARTRSMWSFNPETNILLNVPLHGTILTRPLLESELVIGAVILRGHLRIAGHHLGRCD\n",
      "                     IKDLPKEITVATSRTLSYYKLGASQRVAGDSGFAAYSRYRIGNYKLNTDHSSSSDNIA\n",
      "                     LLVQ\"\n",
      "     gene            27127..27312\n",
      "                     /gene=\"ORF6\"\n",
      "     CDS             27127..27312\n",
      "                     /gene=\"ORF6\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF6 protein\"\n",
      "                     /protein_id=\"WPK49449.1\"\n",
      "                     /translation=\"MFHLVDFQVTIAEILLIIMRTFKVSIWNLDYIINLIIKNLSKSL\n",
      "                     TENKYSQLDEEQPMEIL\"\n",
      "     gene            27319..27684\n",
      "                     /gene=\"ORF7a\"\n",
      "     CDS             27319..27684\n",
      "                     /gene=\"ORF7a\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF7a protein\"\n",
      "                     /protein_id=\"WPK49450.1\"\n",
      "                     /translation=\"MKIILFLALITLATCELYHYQECVRGTTVLLKEPCSSGTYEGNS\n",
      "                     PFHPLADNKFALTCFSTQFAFACPDGVKHVYQLRARSVSPKLFIRQEEVQELYSPIFL\n",
      "                     IVAAIVFITLCFTLKRKTE\"\n",
      "     gene            27681..27812\n",
      "                     /gene=\"ORF7b\"\n",
      "     CDS             27681..27812\n",
      "                     /gene=\"ORF7b\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF7b\"\n",
      "                     /protein_id=\"WPK49451.1\"\n",
      "                     /translation=\"MIELSLIDFYLCFLAFLLFLVLIMLIIFWFSLELQDHNETCHA\"\n",
      "     gene            27819..28184\n",
      "                     /gene=\"ORF8\"\n",
      "     misc_feature    27819..28184\n",
      "                     /gene=\"ORF8\"\n",
      "                     /note=\"similar to ORF8 protein\"\n",
      "     gene            28199..29449\n",
      "                     /gene=\"N\"\n",
      "     CDS             28199..29449\n",
      "                     /gene=\"N\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"nucleocapsid phosphoprotein\"\n",
      "                     /protein_id=\"WPK49452.1\"\n",
      "                     /translation=\"MSDNGPQNQRNALRITFGGPSDSTGSNQNGGARSKQRRPQGLPN\n",
      "                     NTASWFTALTQHGKEDLKFPRGQGVPINTNSSPDDQIGYYRRATRRIRGGDGKMKDLS\n",
      "                     PRWYFYYLGTGPEAGLPYGANKDGIIWVATEGALNTPKDHIGTRNPANNAAIVLQLPQ\n",
      "                     GTTLPKGFYAEGSRGGSQASSRSSSRSRNSSRNSTPGNSKRTSPARMAGNGGDAALAL\n",
      "                     LLLDRLNQLESKMSGKGQQQQGQTVTKKSAAEASKKPRQKRTATKAYNVTQAFGRRGP\n",
      "                     EQTQGNFGDQELIRQGTDYKHWPQIAQFAPSASAFFGMSRIGMEVTPSGTWLTYTGAI\n",
      "                     KLDDKDPNFKDQVILLNKHIDAYKTFPPTEPKKDKKKKADETQALPQRQKKQQTVTLL\n",
      "                     PAADLDDFSKQLQQSMSRADSTQA\"\n",
      "     gene            29474..29590\n",
      "                     /gene=\"ORF10\"\n",
      "     CDS             29474..29590\n",
      "                     /gene=\"ORF10\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF10 protein\"\n",
      "                     /protein_id=\"WPK49453.1\"\n",
      "                     /translation=\"MGYINVFAFPFTIYSLLLCRMNSRNYIAQVDVVNFNLT\"\n",
      "     stem_loop       29525..29560\n",
      "                     /gene=\"ORF10\"\n",
      "                     /note=\"Coronavirus 3' UTR pseudoknot stem-loop 1\"\n",
      "     stem_loop       29545..29573\n",
      "                     /gene=\"ORF10\"\n",
      "                     /note=\"Coronavirus 3' UTR pseudoknot stem-loop 2\"\n",
      "     stem_loop       29644..>29648\n",
      "                     /note=\"Coronavirus 3' stem-loop II-like motif (s2m)\"\n",
      "ORIGIN      \n",
      "        1 agatctgttc tctaaacgaa ctttaaaatc tgtgtggctg tcactcggct gcatgcttag\n",
      "       61 tgcactcacg cagtataatt aataactaat tactgtcgtt gacaggacac gagtaactcg\n",
      "      121 tctatcttct gcaggctgct tacggtttcg tccgtgttgc agccgatcat cagcacatct\n",
      "      181 aggttttgtc cgggtgtgac cgaaaggtaa gatggagagc cttgtccctg gtttcaacga\n",
      "      241 gaaaacacac gtccaactca gtttgcctgt tttacaggtt cgcgacgtgc tcgtacgtgg\n",
      "      301 ctttggagac tccgtggagg aggtcttatc agaggcacgt caacatctta gagatggcac\n",
      "      361 ttgtggctta gtagaagttg aaaaaggcgt tttgcctcaa cttgaacagc cctatgtgtt\n",
      "      421 catcaaacgt tcggatgctc gaactgcacc tcatggtcat gttatggttg agctggtagc\n",
      "      481 agaactcgaa ggcattcagt acggtcgtag tggtgagaca cttggtgtcc ttgtccctca\n",
      "      541 tgtgggcgaa ataccagtgg cttaccgcaa ggttcttctt cgtaagaacg gtaataaagg\n",
      "      601 agctggtggc catagntacg gcgccgatct aaagtcattt gacttaggcg acgagcttgg\n",
      "      661 cactgatcct tatgaagatt ttcaagaaaa ctggaacact aaacatagca gtggtgttac\n",
      "      721 ccgtgaactc atgcgtgagc ttaacggagg ggcatacact cgctatgtcg ataacaactt\n",
      "      781 ctgtggccct gatggctacc ctcttgagtg cattaaagac cttctagcac gtgctggtaa\n",
      "      841 agcttcatgc actttgtccg aacaactgga ctttattgac actaagaggg gtgtatactg\n",
      "      901 ctgccgtgaa catgagcatg aaattgcttg gtacacggaa cgttctgaaa agagctatga\n",
      "      961 attgcagaca ccttttgaaa ttaaattggc aaagaaattt gacaccttca atggggaatg\n",
      "     1021 tccaaatttt gtatttccct taaattccat aatcaagact attcaaccaa gggttgaaaa\n",
      "     1081 gaaaaagctt gatggcttta tgggtagaat tcgatctgtc tatccagttg cgtcaccaaa\n",
      "     1141 tgaatgcaac caaatgtgcc tttcaactct catgaagtgt gatcattgtg gtgaaacttc\n",
      "     1201 atggcagacg ggcgattttg ttaaagccac ttgcgaattt tgtggcactg agaatttgac\n",
      "     1261 taaagaaggt gccactactt gtggttactt accccaaaat gctgttgtta aaatttattg\n",
      "     1321 tccagcatgt cacaattcag aagtaggacc tgagcatagt cttgccgaat accataatga\n",
      "     1381 atctggcttg aaaaccattc ttcgtaaggg tggtcgcact attgcctttg gaggctgtgt\n",
      "     1441 gttctcttat gttggttgcc ataacaagtg tgcctattgg gttccacgtg ctagcgctaa\n",
      "     1501 cataggttgt aaccatacag gtgttgttgg agaaggttcc gaaggtctta atgacaacct\n",
      "     1561 tcttgaaata ctccaaaaag agaaagtcaa catcaatatt gttggtgact ttaaacttaa\n",
      "     1621 tgaagagatc gccattattt tggcatcttt ttctgcttcc acaagtgctt ttgtggaaac\n",
      "     1681 tgtgaaaggt ttggattata aagcattcaa acaaattgtt gaatcctgtg gtaattttaa\n",
      "     1741 agttacaaaa ggaaaagcta aaaaaagtgc ctggaatatt ggtgaacaga aatcaatact\n",
      "     1801 gagtcctctt tatgcatttg catcagaggc tgctcgtgtt gtacgatcaa ttttctcccg\n",
      "     1861 cactcttgaa actgctcaaa attctgtgcg tgttttacag aaggccgcta taacaatact\n",
      "     1921 agatggaatt tcacagtatt cactgagact cattgatgct atgatgttca catctgattt\n",
      "     1981 ggctactaac aatctagttg taatggccta cattacaggt ggtgttgttc agttgacttc\n",
      "     2041 gcagtggcta actaacatct ttggcactgt ttatgaaaaa ctcaaacccg tccttgattg\n",
      "     2101 gcttgaagag aagtttaagg aaggtgtaga gtttcttaga gacggttggg aaattgttaa\n",
      "     2161 atttatctca acctgtgctt gtgaaattgt cggtggacaa attgtcacct gtgcaaagga\n",
      "     2221 aattaaggag agtgttcaga cattctttaa gcttgtaaat aaatttttgg ctttgtgtgc\n",
      "     2281 tgactctatc attattggtg gagctaaact taaagccttg aatttaggtg aaacatttgt\n",
      "     2341 cacgcactca aagggattgt acagaaagtg tgttaaatcc agagaagaaa ctggcctact\n",
      "     2401 catgcctcta aaagccccaa aagaaattat cttcttagag ggagaaacac ttcccacaga\n",
      "     2461 agtgttaaca gaggaagttg tcttgaaaac tggtgattta caaccattag aacaacctac\n",
      "     2521 tagtgaagct gttgaagctc cattggttgg tacaccagtt tgtattaacg ggcttatgtt\n",
      "     2581 gctcgaaatc aaagacacag aaaagtactg tgcccttgca cctaatatga tggtaacaaa\n",
      "     2641 caataccttc acactcaaag gcggtgcacc aacaaaggtt acttttggtg atgacactgt\n",
      "     2701 gatagaagtg caaggttaca agagtgtgaa tatcattttt gaacttgatg aaaggattga\n",
      "     2761 taaagtactt aatgagaagt gctctgccta tacagttgaa ctcggtacag aagtaaatga\n",
      "     2821 gttcgcctgt gttgtggcag atgctgtcat aaaaactttg caaccagtat ctgaattact\n",
      "     2881 tacaccactg ggcattgatt tagatgagtg gagtatggct acatactact tatttgatga\n",
      "     2941 gtctggtgag tttaaattgg cttcacatat gtattgttct ttttaccctc cagatgagga\n",
      "     3001 tgaagaagaa ggtgattgtg aagaagaaga gtttgagcca ttaactcaat atgagtatgg\n",
      "     3061 tactgaagat gattaccaag gtaaaccttt ggaatttggt gccacttctg ctgctcttca\n",
      "     3121 acctgaagaa gagcaagaag aagattggtt agatgatgat agtcaacaaa ctgttggtca\n",
      "     3181 acaagacggc agtgaggaca atcagacaac tactattcaa acaattgttg aggttcaacc\n",
      "     3241 tcaattagag atggaactta caccagttgt tcagactatt gaagtgaata gttttagtgg\n",
      "     3301 ttatttaaaa cttactgaca atgtatacat taaaaatgca gacattgtgg aagaagctaa\n",
      "     3361 aaaggtaaaa ccaacagtgg ttgttaatgc agccaatgtt taccttaaac atggaggagg\n",
      "     3421 tgttgcagga gccttaaata aggctactaa caatgccatg caagttgaat ctgatgatta\n",
      "     3481 catagctact aatggaccac ttaaagtggg tgatagttgt gttttaagcg gacacaatct\n",
      "     3541 tgctaaacac tgtcttcatg ttgtcggccc aaatgttaac aaaggtgaag acattcaact\n",
      "     3601 tcttaagagt gcttatgaaa attttaatca gcacgaagtt ctacttgcac cattattatc\n",
      "     3661 agctggtatt tttggtgctg actctataca ttctttaaga gtttgtgtag atactgttcg\n",
      "     3721 cacaaatgtc tacttagctg tctttgataa aaatctctat gacaaacttg tttcaagctt\n",
      "     3781 tttggaaatg aagagtgaaa agcaagttga acaaaagatc gctgagattc ctaaagagga\n",
      "     3841 agttaagcca tttataactg aaagtaaacc ttcagttgaa cagagaaaac aagatgataa\n",
      "     3901 gaaaatcaaa gcttgtgttg aagaagttac aacaactctg gaagaaacta agttcctcac\n",
      "     3961 agaaaacttg ttactttata ttgacattaa tggcaatctt catccagatt ctgccactct\n",
      "     4021 tgttagtgac attgacatca ctttcttaaa gaaagatgct ccatatatag tgggtgatgt\n",
      "     4081 tgttcaagag ggtgttttaa ctgctgtggt tatacctact aaaaaggcta gtggcactac\n",
      "     4141 tgaaatgcta gcgaaagctt tgagaaaagt gccaacagac aattatataa ccacttaccc\n",
      "     4201 gggtcagggt ttaaatggtt acactgtaga ggaggcaaag acagtgctta aaaagtgtaa\n",
      "     4261 aagtgctttt tacattctac catctattat ctctaatgag aagcaagaaa ttcttggaac\n",
      "     4321 tgtttcttgg aatttgcgag aaatgcttgc acatgcagaa gaaacacgca aattaatgcc\n",
      "     4381 tgtctgtgtg gaaactaaag ccatagtttc aactatacag cgtaaatata agggtattaa\n",
      "     4441 aatacaagag ggtgtggttg attatggtgc tagattttac ttttacacca gtaaaacaac\n",
      "     4501 tgtagcgtca cttatcaaca cacttaacga tctaaatgaa actcttgtta caatgccact\n",
      "     4561 tggctatgta acacatggct taaatttgga agaagctgct cggtatatga gatctctcaa\n",
      "     4621 agtgccagct acagtttctg tttcttcacc tgatgctgtt acagcgtata atggttatct\n",
      "     4681 tacttcttct tctaaaacac ctgaagaaca ttttattgaa accatctcac ttgctggttc\n",
      "     4741 ctataaagat tggtcctatt ctggacaatc tacacaacta ggtatagaat ttcttaagag\n",
      "     4801 aggtgataaa agtgtatatt acactagtaa tcctaccaca ttccacctag atggtgaagt\n",
      "     4861 tatcaccttt gacaatctta agacacttct ttctttgaga gaagtgagga ctattaaggt\n",
      "     4921 gtttacaaca gtagacaaca ttaacctcca cacgcaagtt gtggacatgt caatgacata\n",
      "     4981 tggacaacag tttggtccaa cttatttgga tggagctgat gttactaaaa taaaacctca\n",
      "     5041 taattcacat gaaggtaaaa cattttatgt tttacctaat gatgacactc tacgtgttga\n",
      "     5101 ggcttttgag tactaccaca caactgatcc tagttttctg ggtaggtaca tgtcagcatt\n",
      "     5161 aaatcacact aaaaagtgga aatacccaca agttaatggt ttaacttcta ttaaatgggc\n",
      "     5221 agataacaac tgttatcttg ccactgcatt gttaacactc caacaaatag agttgaagtt\n",
      "     5281 taatccacct gctctacaag atgcttatta cagagcaagg gctggtgaag ctgctaactt\n",
      "     5341 ttgtgcactt atcttagcct actgtaataa gacagtaggt gagttaggtg atgttagaga\n",
      "     5401 aacaatgagt tacttgtttc aacatgccaa tttagattct tgcaaaagag tcttgaacgt\n",
      "     5461 ggtgtgtaaa acttgtggac aacagcagac aacccttaag ggtgtagaag ctgttatgta\n",
      "     5521 catgggcaca ctttcttatg aacaatttaa gaaaggtgtt cagatacctt gtacgtgtgg\n",
      "     5581 taaacaagct acaaaatatc tagtacaaca ggagtcacct tttgttatga tgtcagcacc\n",
      "     5641 acctgctcag tatgaactta agcatggtac atttacttgt gctagtgagt acactggtaa\n",
      "     5701 ttaccagtgt ggtcactata aacatataac ttctaaagaa actttgtatt gcatagacgg\n",
      "     5761 tgctttactt acaaagtcct cagaatacaa aggtcctatt acggatgttt tctacaaaga\n",
      "     5821 aaacagttac acaacaacca taaaaccagt tacttataaa ttggatggtg ttgtttgtac\n",
      "     5881 agaaattgac cctaagttgg acaattatta taagaaagac aattcttatt tcacagagca\n",
      "     5941 accaattgat cttgtaccaa accaaccata tccaaacgca agcttcgata attttaagtt\n",
      "     6001 tgtatgtgat aatatcaaat ttgctgatga tttaaaccag ttaactggtt ataagaaacc\n",
      "     6061 tgcttcaaga gagcttaaag ttacattttt ccctgactta aatggtgatg tggtggctat\n",
      "     6121 tgattataaa cactacacac cctcttttaa gaaaggagct aaattgttac ataaacctat\n",
      "     6181 tgtttggcat gttaacaatg caactaataa agccacgtat aaaccaaata cctggtgtat\n",
      "     6241 acgttgtctt tggagcacaa aaccagttga aacatcaaat tcgtttgatg tactgaagtc\n",
      "     6301 agaggacgcg cagggaatgg ataatcttgc ctgcgaagat ctaaaaccag tctctgaaga\n",
      "     6361 agtagtggaa aatcctacca tacagaaaga cgttcttgag tgtaatgtga aaactaccga\n",
      "     6421 agttgtagga gacattatac ttaaaccagc aaataatagt ttaaaaatta cagaagaggt\n",
      "     6481 tggccacaca gatctaatgg ctgcttatgt agacaattct agtcttacta ttaagaaacc\n",
      "     6541 taatgaatta tctagagtat taggtttgaa aacccttgct actcatggtt tagctgctgt\n",
      "     6601 taatagtgtc ccttgggata ctatagctaa ttatgctaag ccttttctta acaaagttgt\n",
      "     6661 tagtacaact actaacatag ttacacggtg tttaaaccgt gtttgtacta attatatgcc\n",
      "     6721 ttatttcttt actttattgc tacaattgtg tacttttact agaagtacaa attctagaat\n",
      "     6781 taaagcatct atgccgacta ctatagcaaa gaatactgtt aagagtgtcg gtaaattttg\n",
      "     6841 tctagaggct tcatttaatt atttgaagtc acctaatttt tctaaactga taaatattat\n",
      "     6901 aatttggttt ttactattaa gtgtttgcct aggttcttta atctactcaa ccgctgcttt\n",
      "     6961 aggtgtttta atgtctaatt taggcatgcc ttcttactgt actggttaca gagaaggcta\n",
      "     7021 tttgaactct actaatgtca ctattgcaac ctactgtact ggttctatac cttgtagtgt\n",
      "     7081 ttgtcttagt ggtttagatt ctttagacac ctatccttct ttagaaacta tacaaattac\n",
      "     7141 catttcatct tttaaatggg atttaactgc ttttggctta gttgcagagt ggtttttggc\n",
      "     7201 atatattctt ttcactaggt ttttctatgt acttggattg gctgcaatca tgcaattgtt\n",
      "     7261 tttcagctat tttgcagtac attttattag taattcttgg cttatgtggt taataattaa\n",
      "     7321 tcttgtacaa atggccccga tttcagctat ggttagaatg tacatcttct ttgcatcatt\n",
      "     7381 ttattatgta tggaaaagtt atgtgcatgt tgtagacggt tgtaattcat caacttgtat\n",
      "     7441 gatgtgttac aaacgtaata gagcaacaag agtcgaatgt acaactattg ttaatggtgt\n",
      "     7501 tagaaggtcc ttttatgtct atgctaatgg aggtaaaggc ttttgcaaac tacacaattg\n",
      "     7561 gaattgtgtt aattgtgata cattctgtgc tggtagtaca tttattagtg atgaagttgc\n",
      "     7621 gagagacttg tcactacagt ttaaaagacc aataaatcct actgaccagt cttcttacat\n",
      "     7681 cgttgatagt gttacagtga agaatggttc catccatctt tactttgata aagctggtca\n",
      "     7741 aaagacttat gaaagacatt ctctctctca ttttgttaac ttagacaacc tgagagctaa\n",
      "     7801 taacactaaa ggttcattgc ctattaatgt tatagttttt gatggtaaat caaaatgtga\n",
      "     7861 agaatcatct gcaaaatcag cgtctgttta ctacagtcag cttatgtgtc aacctatact\n",
      "     7921 gttactagat caggcattag tgtctgatgt tggtgatagt gcggaagttg cagttaaaat\n",
      "     7981 gtttgatgct tacgttaata cgttttcatc aacttttaac gtaccaatgg aaaaactcaa\n",
      "     8041 aacactagtt gcaactgcng aagctgaact tgcaaagaat gtgtccttag acaatgtctt\n",
      "     8101 atctactttt atttcagcag ctcggcaagg gtttgttgat tcagatgtag aaactaaaga\n",
      "     8161 tgttgttgaa tgtcttaaat tgtcacatca atctgacata gaagttactg gcgatagttg\n",
      "     8221 taataactat atgctcacct ataacaaagt tgaaaacatg acaccccgtg accttggtgc\n",
      "     8281 ttgtattgac tgtagtgcgc gtcatattaa tgcgcaggta gcaaaaagtc acaacattgc\n",
      "     8341 tttgatatgg aacgttaaag atttcatgtc attgtctgaa caactacgaa aacaaatacg\n",
      "     8401 tagtgctgct aaaaagaata acttaccttt taagttgaca tgtgcaacta ctagacaagt\n",
      "     8461 tgttaatgtt gtaacaacaa agatagcact taagggtggt aaaattgtta ataattggtt\n",
      "     8521 gaagcagtta attaaagtta cacttgtgtt cctttttgtt gctgctattt tctatttaat\n",
      "     8581 aacacctgtt catgtcatgt ctaaacatac tgacttttca agtgaaatca taggatacaa\n",
      "     8641 ggctattgat ggtggtgtca ctcgtgacat agcatctaca gatacttgtt ttgctaacaa\n",
      "     8701 acatgctgat tttgacacat ggtttagcca gcgtggtggt agttatacta atgacaaagc\n",
      "     8761 ttgcccattg attgctgcag tcataacaag agaagtgggt tttgtcgtgc ctggtttgcc\n",
      "     8821 tggcacgata ttacgcacaa ctaatggtga ctttttgcat ttcttaccta gagtttttag\n",
      "     8881 tgcagttggt aacatctgtt acacaccatc aaaacttata gagtacactg actttgcaac\n",
      "     8941 atcagcttgt gttttggctg ctgaatgtac aatttttaaa gatgcttctg gtaagccagt\n",
      "     9001 accatattgt tatgatacca atgtactaga aggttctgtt gcttatgaaa gtttacgccc\n",
      "     9061 tgacacacgt tatgtgctca tggatggctc tattattcaa tttcctaaca cctaccttga\n",
      "     9121 aggttctgtt agagtggtaa caacttttga ttctgagtac tgtaggcacg gcacttgtga\n",
      "     9181 aagatcagaa gctggtgttt gtgtatctac tagtggtaga tgggtactta acaatgatta\n",
      "     9241 ttacagatct ttaccaggag ttttttgtgg tgtagatgct gtaaatttat ttactaatat\n",
      "     9301 gtttacacca ctaattcaac ctattggtgc tttggacata tcagcatcta tagtagctgg\n",
      "     9361 tggtattgtg gctatcgtag taacatgcct tgcctactat tttatgaggt ttagaagagc\n",
      "     9421 ttttggtgaa tacagtcatg tagttgcctt taatacttta ctattcctta tgtcattcat\n",
      "     9481 tgtactctgt ttaacaccag tttactcatt cttacctggt gtttattctg ttatttactt\n",
      "     9541 gtacttgaca ttttatctta ctaatgatgt ttctttttta gcacatattc agtggatggt\n",
      "     9601 tatgttcaca cctttagtac ctttctggat aacaattgct tatatcattt gtatttccac\n",
      "     9661 aaagcatttc tattggttct ttagtaatta cctaaagaga cgtgtagtct ttaatggtgt\n",
      "     9721 ttcctttagt acttttgaag aagctgcgct gtgcaccttt ttgttaaata aagaaatgta\n",
      "     9781 tctaaagttg cgtagtgatg tgctattacc ttttacgcaa tataatagat acttagctct\n",
      "     9841 ttataataag tacaagtatt ttagtggagc aatggataca actagctaca gagaagctgc\n",
      "     9901 ttgttgtcat ctcgcaaagg ctctcaatga cttcagtaac tcaggttctg atgttcttta\n",
      "     9961 ccaannnnnn nnnnnntcta tcacctcagc tgttttgcag agtggtttta gaaaaatggc\n",
      "    10021 attcccatct ggtaaagttg agnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    10081 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    10141 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    10201 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    10261 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnttca\n",
      "    10321 accaggacag actttttcag tgttagcttg ttacaatggt tcaccatctg gtgtttacca\n",
      "    10381 atgtgctatg agacacaatt tcactattaa gggttcattc cttaatggtt catgtggtag\n",
      "    10441 tgttggtttt aacatagatt atgactgtgt ctctttttgt tacatgcacc atatggaatt\n",
      "    10501 accaactgga gttcatgctg gcacagactt agaaggtaac ttttatggac cttttgttga\n",
      "    10561 caggcaaaca gcacaagcag ctggtacgga cacaactatt acagttaatg ttttagcttg\n",
      "    10621 gttgtacgct gctgttataa atggagacag gtggtttctc aatcgattta ccacaactct\n",
      "    10681 taatgacttt aaccttgtgg ctatgaagta caattatgaa cctctaacac aagaccatgt\n",
      "    10741 tgacatacta ggacctcttt ctgctcaaac tggaattgcc gttttagata tgtgtgcttc\n",
      "    10801 attaaaagaa ttactgcaaa atggtatgaa tggacgtacc atattgggta gtgctttatt\n",
      "    10861 agaagatgaa tttacacctt ttgatgttgt tagacaatgc tcaggtgtta ctttccaaag\n",
      "    10921 tgcagtgaaa agaacaatca agggtacaca ccactggttg ttactcacaa ttttgacttc\n",
      "    10981 acttttagtt ttagtccaga gtactcaatg gtctttgttc ttttttttgt atgaaaatgc\n",
      "    11041 ctttttacct tttgctatgg gtattattgc tatgtctgct tttgcaatga tgtttgtcaa\n",
      "    11101 acataagcat gcatttctct gtttgttttt gttaccttct cttgccactg tagcttattt\n",
      "    11161 taatatggtc tatatgcctg ctagttgggt gatgcgtatt atgacatggt tggatatggt\n",
      "    11221 tgatactagt ttgaagctaa aagactgtgt tatgtatgca tcagctgtag tgttactaat\n",
      "    11281 ccttatgaca gcaagaactg tgtatgatga tggtgctagg agagtgtgga cacttatgaa\n",
      "    11341 tgtcttgaca ctcgtttata aagtttatta tggtaatgct ttagatcaag ccatttccat\n",
      "    11401 gtgggctctt ataatctctg ttacttctaa ctactcaggt gtagttacaa ctgtcatgtt\n",
      "    11461 tttggccaga ggtattgttt ttatgtgtgt tgagtattgc cctattttct tcataactgg\n",
      "    11521 taatacactt cagtgtataa tgctagttta ttgtttctta ggctattttt gtacttgtta\n",
      "    11581 ctttggcctc ttttgtttac tcaaccgcta ctttagactg actcttggtg tttatgatta\n",
      "    11641 cttagtttct acacaggagt ttagatatat gaattcacag ggactattcc cacccaagaa\n",
      "    11701 tagcatagat gccttcaaac tcaacattaa attgttgggt gttggtggca aaccttgtat\n",
      "    11761 caaagtagcc actgtacagt ctaaaatgtc agatgtaaag tgcacatcag tagtcttact\n",
      "    11821 ctcagttttg caacaactca gagtagaatc atcatctaaa ttgtgggctc aatgtgtcca\n",
      "    11881 gttacacaat gatattctct tagctaaaga tactactgaa gcctttgaaa aaatggtttc\n",
      "    11941 actactttct gttttgcttt ccatgcaggg tgctgtagac ataaacaagc tttgtgaaga\n",
      "    12001 aatgctggac aacagggcaa ccttacaagc tatagcctca gagtttagtt cccttccatc\n",
      "    12061 atatgcagct tttgctactg ctcaagaagc ttatgagcag gctgttgcta atggtgattc\n",
      "    12121 tgaagttgtt cttaaaaagt tgaagaagtc tttgaatgtg gctaaatctg aatttgaccg\n",
      "    12181 tgatgcagcc atgcaacgta agttggaaaa gatggctgat caagctatga cccaaatgta\n",
      "    12241 taaacaggct agatctgagg acaagagggc aaaagttact agtgctatgc agacaatgct\n",
      "    12301 tttcactatg cttagaaagt tggataatga tgcactcaac aacattatca acaatgcaag\n",
      "    12361 agatggttgt gttcccttga acataatacc tcttacaaca gcagccaaac taatggttgt\n",
      "    12421 cataccagac tataacacat ataaaaatac gtgtgatggt acaacattta cttatgcatc\n",
      "    12481 agcattgtgg gaaatccaac aggttgtaga tgcagatagt aaaattgttc aacttagtga\n",
      "    12541 aattagtatg gacaattcac ctaatttagc atggcctctt attgtaacag ctttaagggc\n",
      "    12601 caattctgct gtcaaattac agaataatga gcttagtcct gttgcactac gacagatgtc\n",
      "    12661 ttgtgcagcc ggtactacac aaactgcttg cactgatgac aatgcgttag cttactacaa\n",
      "    12721 cacaacaaag ggaggtaggt ttgtacttgc actgttatcc gatttacagg atttgaaatg\n",
      "    12781 ggctagattc cctaagagtg atggaactgg tactatttat acagaactgg aaccaccttg\n",
      "    12841 taggtttgtt acagacacac ctaaaggtcc taaagtgaag tatttatact ttattaaagg\n",
      "    12901 attaaacaac ctaaatagag gtatggtact tggtagttta gctgccacag tacgtctaca\n",
      "    12961 agctggtaat gcaacagaag tgcctgccaa ttcaactgta ttatctttct gtgcttttgc\n",
      "    13021 tgtagatgct gctaaagctt acaaagatta tctagctagt gggggacaac caatcactaa\n",
      "    13081 ttgtgttaag atgttgtgta cacacactgg tactggtcag gcaataacag ttacaccgga\n",
      "    13141 agccaatatg gatcaagaat cctttggtgg tgcatcgtgt tgtctgtact gccgttgcca\n",
      "    13201 catagatcat ccaaatccta aaggattttg tgacttaaaa ggtaagtatg tacaaatacc\n",
      "    13261 tacaacttgt gctaatgacc ctgtgggttt tacacttaaa aacacagtct gtaccgtctg\n",
      "    13321 cggtatgtgg aaaggttatg gctgtagttg tgatcaactc cgcgaaccca tgcttcagtc\n",
      "    13381 agctgatgca caatcgtttt taaacgggtt tgcggtgtaa gtgcagcccg tcttacaccg\n",
      "    13441 tgcggcacag gcactagtac tgatgtcgta tacagggctt ttgacatcta caatgataaa\n",
      "    13501 gtagctggtt ttgctaaatt cctaaaaact aattgttgtc gcttccaaga aaaggacgaa\n",
      "    13561 gatgacaatt taattgattc ttactttgta gttaagagac acactttctc taactaccaa\n",
      "    13621 catgaagaaa caatttataa tttacttaag gattgtctag ctgttgctaa acatgacttc\n",
      "    13681 tttaagttta gaatagacgg tgacatggta ccacatatat cacgtcaacg tcttactaaa\n",
      "    13741 tacacaatgg cagacctcgt ctatgcttta aggcattttg atgaaggtaa ttgtgacaca\n",
      "    13801 ttaaaagaaa tacttgtcac atacaattgt tgtgatgatg attatttcaa taaaaaggac\n",
      "    13861 tggtatgatt ttgtagaaaa cccagatata ttacgcgtat acgccaactt aggtgaacgt\n",
      "    13921 gtacgccaag ctttgttaaa aacagtacaa ttctgtgatg ccatgcgaaa tgctggtatt\n",
      "    13981 gttggtgtac tgacattaga taatcaagat ctcaatggta actggtatga tttcggtgat\n",
      "    14041 ttcatacaaa ccacgccagg tagtggagtt cctgttgtag attcttatta ttcattgtta\n",
      "    14101 atgcctatat taaccttgac cagggcttta actgcagagt cacatgttga cactgactta\n",
      "    14161 acaaagcctt acattaagtg ggatttgtta aaatatgact tcacggaaga gaggttaaaa\n",
      "    14221 ctctttgacc gttattttaa atattgggat cagacatacc acccaaattg tgttaactgt\n",
      "    14281 ttggatgaca gatgcattct gcattgtgca aactttaatg ttttattctc tacagtgttc\n",
      "    14341 ccacttacaa gttttggacc actagtgaga aaaatatttg ttgatggtgt tccatttgta\n",
      "    14401 gtttcaactg gataccactt cagagagcta ggtgttgtac ataatcagga tgtaaactta\n",
      "    14461 catagctcta gacttagttt taaggaatta cttgtgtatg ctgctgaccc tgctatgcac\n",
      "    14521 gctgcttctg gtaatctatt actagataaa cgcactacgt gcttttcagt agctgcactt\n",
      "    14581 actaacaatg ttgcttttca aactgtcaaa cccggtaatt ttaacaaaga cttctatgac\n",
      "    14641 tttgctgtgt ctaagggttt ctttaaggaa ggaagttctg ttgaattaaa acacttcttc\n",
      "    14701 tttgctcagg atggtaatgc tgctatcagc gattatgact actatcgtta taatctacca\n",
      "    14761 acaatgtgtg atatcagaca actactattt gtggttgaag ttgttgataa gtactttgat\n",
      "    14821 tgttacgatg gtggctgtat taatgctaac caagtcatcg tcaacaacct agacaaatca\n",
      "    14881 gctggttttc catttaataa atggggtaag gctagacttt attatgattc aatgagttat\n",
      "    14941 gaggatcaag atgcactttt cgcatataca aaacgtaatg tcatccctac tataactcaa\n",
      "    15001 atgaatctta agtatgccat tagtgcaaag aatagagctc gcaccgtagc tggtgtctct\n",
      "    15061 atctgtagta ctatgaccaa tagacagttt catcaaaaat tattgaaatc aatagccgcc\n",
      "    15121 actagaggag ctactgtagt aattggaaca agcaaattct atggtggttg gcacaacatg\n",
      "    15181 ttaaaaactg tttatagtga tgtagaaaac cctcacctta tgggttggga ttatcctaaa\n",
      "    15241 tgtgatagag ccatgcctaa catgcttaga attatggcct cacttgttct tgctcgcaaa\n",
      "    15301 catacaacgt gttgtagctt gtcacaccgt ttctatagat tagctaatga gtgtgctcaa\n",
      "    15361 gtattgagtg aaatggtcat gtgtggcagt tcactatatg ttaaaccagg tggaacctca\n",
      "    15421 tcaggagatg ccacaactgc ttatgctaat agtgttttta acatttgtca agctgtcacg\n",
      "    15481 gccaatgtta atgcactttt atctactgat ggtaacaaaa ttgccgataa gtatgtccgc\n",
      "    15541 aatttacaac acagacttta tgagtgtctc tatagaaata gagatgttga cacagacttt\n",
      "    15601 gtgaatgagt tttacgcata tttgcgtaaa catttctcaa tgatgatact ttctgacgat\n",
      "    15661 gctgttgtgt gttttaatag cacttatgca tctcaaggtc tagtggctag cataaagaac\n",
      "    15721 tttaagtcag ttctttatta tcaaaacaat gtttttatgt ctgaagcaaa atgttggact\n",
      "    15781 gagactgacc ttactaaagg acctcatgaa ttttgctctc aacatacaat gctagttaaa\n",
      "    15841 cagggtgatg attatgtgta ccttccttac ccagacccat caagaatcct aggggccggc\n",
      "    15901 tgttttgtag atgatatcgt aaaaacagat ggtacactta tgattgaacg gttcgtgtct\n",
      "    15961 ttagctatag atgcttaccc acttactaaa catcctaatc aggagtatgc tgatgtcttt\n",
      "    16021 catttgtact tacaatacat aagaaagcta catgatgagt taacaggaca catgttagac\n",
      "    16081 atgtattctg ttatgcttac taatgataac acttcaaggt attgggaacc tgagttttat\n",
      "    16141 gaggctatgt acacaccgca tacagtctta caggctgttg gggcttgtgt tctttgcaat\n",
      "    16201 tcacagactt cattaagatg tggtgcttgc atacgtagac cattcttatg ttgtaaatgc\n",
      "    16261 tgttacgacc atgtcatacc aacatcacat aaattagtct tgtctgttaa tccgtatgtt\n",
      "    16321 tgcaatgccc caggttgtga tgtcacagat gtgactcaac tttacttagg aggtatgagc\n",
      "    16381 tattattgta aatcacataa accacccatt agttttccat tgtgtgctaa tggacaagtt\n",
      "    16441 tttggtttat ataaaaatac atgtgttggt agcgataatg ttactgactt taatgcaatt\n",
      "    16501 gcaacatgtg actggacaaa tgctggtgat tacattttag ctaacacctg tactgaaaga\n",
      "    16561 ctcaagcttt ttgcagcaga aacgctcaaa gctactgagg agacatttaa actgtcttat\n",
      "    16621 ggtattgcta ctgtacgtga agtgctgtct gacagagaat tacatctttc atgggaagtt\n",
      "    16681 ggtaaaccta gaccaccact taaccgaaat tatgtcttta ctggttatcg tgtaactaaa\n",
      "    16741 aacagtaaag tacaaatagg agagtacacc tttgaaaaag gtgactatgg tgatgctgtt\n",
      "    16801 gtttaccgag gtacaacaac ttacaaatta aatgttggtg attattttgt gctgacatca\n",
      "    16861 catacagtaa tgccattaag tgcacctaca ctagtgccac aagagcacta tgttagaatt\n",
      "    16921 actggcttat acccaacact caatatctca gatgagtttt ctagcaatgt tgcaaattat\n",
      "    16981 caaaaggttg gtatgcaaaa gtattctaca ctccagggac cacctggtac tggtaagagt\n",
      "    17041 cattttgcta ttggcctagc tctctactac ccttctgctc gcatagtgta tacagcttgc\n",
      "    17101 tctcatgccg ctgttgatgc actatgtgag aaggcattaa aatatttgcc tatagataaa\n",
      "    17161 tgtagtagaa ttatacctgc acgtgctcgt gtagagtgtt ttgataaatt caaagtgaat\n",
      "    17221 tcaacattag aacagtatgt cttttgtact gtaaatgcat tgcctgagac gacagcagat\n",
      "    17281 atagttgtct ttgatgaaat ttcaatggcc acaaattatg atttgagtgt tgtcaatgcc\n",
      "    17341 agattatgtg ctaagcacta tgtgtacatt ggcgaccctg ctcaattacc tgcaccacgc\n",
      "    17401 acattgctaa ctaagggcac actagaacca gaatatttca attcagtgtg tagacttatg\n",
      "    17461 aaaactatag gtccagacat gttcctcgga acttgtcggc gttgtcctgc tgaaattgtt\n",
      "    17521 gacactgtga gtgctttggt ttatgataat aagcttaaag cacataaaga caaatcagct\n",
      "    17581 caatgcttta aaatgtttta taagggtgtt atcacgcatg atgtttcatc tgcaattaac\n",
      "    17641 aggccacaaa taggcgtggt aagagaattc cttacacgta accctgcttg gagaaaagct\n",
      "    17701 gtctttattt caccttataa ttcacagaat gctgtagcct caaagatttt gggactacca\n",
      "    17761 actcaaactg ttgattcatc acagggctca gaatacgact atgtcatatt cactcaaacc\n",
      "    17821 actgaaacag ctcactcttg taatgtaaac agatttaatg ttgctattac cagagcaaaa\n",
      "    17881 gtaggtatac tttgcataat gtctgataga gacctttatg acaagttgca atttacaagt\n",
      "    17941 cttgaaattc cacgtaggaa tgtggcaact ttacaagctg aaaatgtaac aggactcttt\n",
      "    18001 aaagattgta gtaaggtaat cactgggtta catcctacac aggcacctac acacctcagt\n",
      "    18061 gttgacacta aattcaaaac tgaaggttta tgtgttgacg tacctggcat acctaaggac\n",
      "    18121 atgacctata gaagactcat ctctatgatg ggttttaaaa tgaattatca agttaatggt\n",
      "    18181 taccctaaca tgtttatcac ccgcgaagaa gctataagac atgtacgtgc atggattggc\n",
      "    18241 ttcgatgtcg aggggtgtca tgctactaga gaagctgttg gtaccaattt acctttacag\n",
      "    18301 ctaggttttt ctacaggtgt taacctagtt gctgtaccta caggttatgt tgatacacct\n",
      "    18361 aataatacag atttttccag agttagtgct aaaccaccgc ctggagatca atttaaacac\n",
      "    18421 ctcataccac ttatgtacaa aggacttcct tggaatgtag tgcgtataaa gattgtacaa\n",
      "    18481 atgttaagtg acacacttaa aaatctctct gacagagtcg tatttgtctt atgggcacat\n",
      "    18541 ggctttgagt tgacatctat gaagtatttt gtgaaaatag gacctgagcg cacctgttgt\n",
      "    18601 ctatgtgata gacgtgccac atgcttttcc actgcttcat acacttatgc ctgttggcat\n",
      "    18661 cattctattg gatttgatta cgtctataat ccgtttatga ttgatgttca acaatggggt\n",
      "    18721 tttacaggta acctacaaag caaccatgat ctgtattgtc aagtccatgg taatgcacat\n",
      "    18781 gtagctagtt gtgatgcaat catgactagg tgtctagctg tccacgagtg ctttgttaag\n",
      "    18841 cgtgttgact ggactattga atatcctata attggtgatg aactgaagat taatgcggct\n",
      "    18901 tgtagaaagg ttcaacacat ggttgttaaa gctgcattat tagcagacaa attcccagtt\n",
      "    18961 cttcacgaca ttggtaaccc taaagctatt aagtgtgtac ctcaagctga tgtagaatgg\n",
      "    19021 aagttctatg atgcacagcc ttgtagtgac aaagcttata aaatagaaga attattctat\n",
      "    19081 tcttatgcca cacattctga caaattcaca gatggtgtat gcctattttg gaattgcaat\n",
      "    19141 gtcgatagat atcctgctaa ttccattgtt tgtagatttg acactagagt gctatctaac\n",
      "    19201 cttaacttgc ctggttgtga tggtggcagt ttgtatgtaa ataaacatgc attccacaca\n",
      "    19261 ccggcttttg ataaaagtgc ttttgttaat ttaaaacaat taccattttt ctattactct\n",
      "    19321 gacagtccat gtgagtctca tggaaaacaa gtagtgtcag atatagatta tgtaccacta\n",
      "    19381 aagtctgcta cgtgtataac acgttgcaat ttaggtggtg ctgtctgtag acatcatgct\n",
      "    19441 aatgagtaca gattgtatct cgatgcttat aacatgatga tctcagctgg ctttagcttg\n",
      "    19501 tgggtttaca aacaatttga tacttataac ctctggaaca cttttacaag acttcagagt\n",
      "    19561 ttagaaaatg tggcttttaa tgttgtaaat aagggacact ttgatggaca acagggtgaa\n",
      "    19621 gtaccagttt ctatcattaa taacactgtt tacacaaaag ttgatggtgt tgatgtagaa\n",
      "    19681 ttgtttgaaa ataaaacaac attacctgtt aatgtagcat ttgagctttg ggctaagcgc\n",
      "    19741 aacattaaac cagtaccaga ggtgaaaata ctcaataatt tgggtgtgga cattgctgct\n",
      "    19801 aatactgtga tctgggacta caaaagagat gctccagcac atatatctac tattggtgtt\n",
      "    19861 tgttctatga ctgacatagc caagaaacca attgaaacga tttgtgcacc actcactgtc\n",
      "    19921 ttttttgatg gtagagttga tggtcaagta gacttattta gaaatgcccg taatggtgtt\n",
      "    19981 cttattacag agggtagtgt taaaggttta caaccatctg taggtcccaa acaagctagt\n",
      "    20041 cttaatggag tcacattaat tggagaagcc gtaaaaacac agttcaatta ttataagaaa\n",
      "    20101 gttgatggtg ttgtccaaca attacctgaa acttacttta ctcagagtag aaatttacaa\n",
      "    20161 gaatttaaac ccaggagtca aatggaaatt gatttcttag aattagctat ggatgaattc\n",
      "    20221 attgaacggt ataaattaga aggctatgcc ttcgaacata tcgtttatgg agattttagt\n",
      "    20281 catagtcagt taggtggttt acatctactg attggactag ctaaacgttt taaggaatca\n",
      "    20341 ccttttgaat tagaagattt tattcctatg gacagtacag ttaaaaacta tttcataaca\n",
      "    20401 gatgcgcaaa caggttcatc taagtgtgtg tgttctgtta ttgatttatt acttgatgat\n",
      "    20461 tttgttgaaa taataaaatc ccaagattta tctgtagttt ctaaggttgt caaagtgact\n",
      "    20521 attgactata cagaaatttc atttatgctt tggtgtaaag atggccatgt agaaacattt\n",
      "    20581 tacccaaaat tacaatctag tcaagcgtgg caaccgggtg ttgctatgcc taatctttac\n",
      "    20641 aaaatgcaaa gaatgctatt agaaaagtgt gaccttcaaa attatggtga tagtgcaaca\n",
      "    20701 ttacctaaag gcataatgat gaatgtcgca aaatatactc aactgtgtca atatttaaac\n",
      "    20761 acattaacat tagctgtacc ctataatatg agagttatac attttggtgc tggttctgat\n",
      "    20821 aaaggagttg caccaggtac agctgtttta agacagtggt tgcctacggg tacgctgctt\n",
      "    20881 gtcgattcag atcttaatga ctttgtctct gatgcagatt caactttgat tggtgattgt\n",
      "    20941 gcaactgtac atacagctaa taaatgggat ctcattatta gtgatatgta cgaccctaag\n",
      "    21001 actaaaaatg ttacaaaaga aaatgactct aaagagggtt ttttcactta catttgtggg\n",
      "    21061 tttatacaac aaaagctagc tcttggaggt tccgtggcta taaagataac agaacattct\n",
      "    21121 tggaatgctg atctttataa gctcatggga cacttcgcat ggtggacagc ctttgttact\n",
      "    21181 aatgtgaatg cgtcatcatc tgaagcattt ttaattggat gtaattatct tggcaaacca\n",
      "    21241 cgcgaacaaa tagatggtta tgtcatgcat gcaaattaca tattttggag gaatacaaat\n",
      "    21301 ccaattcagt tgtcttccta ttctttattt gacatgagta aatttcccct taaattaagg\n",
      "    21361 ggtactgctg ttatgtcttt aaaagaaggt caaatcaatg atatgatttt atctcttctt\n",
      "    21421 agtaaaggta gacttataat tagagaaaac aacagagttg ttatttctag tgatgttctt\n",
      "    21481 gttaacaact aaacgaacaa tgtttgtttt tcttgtttta ttgccactag tctctagtca\n",
      "    21541 gtgtgttaat cttataacca gaactcaatc atacactaat tctttcacac gtggtgttta\n",
      "    21601 ttaccctgac aaagttttca gatcctcagt tttacattca actcaggact tgttcttacc\n",
      "    21661 tttcttttcc aatgttactt ggttccatgc tatacatgtc tctgggacca atggtactaa\n",
      "    21721 gaggtttgat aaccctgccc taccatttaa tgatggtgtt tattttgctt ccactgagaa\n",
      "    21781 gtctaacata ataagaggct ggatttttgg tactacttta gattcgaaga cccagtccct\n",
      "    21841 acttattgtt aataacgcta ctaatgttgt tattaaagtc tgtgaatttc aattttgtaa\n",
      "    21901 tgatccattt ttggatgttt accaaaaaaa caacaaaagt tggatggaaa gtgagttcag\n",
      "    21961 agtttattct agtgcgaata attgcacttt tgaatatgtc tctcagcctt ttcttatgga\n",
      "    22021 ccttgtagga aaagagggta atttcaaaaa tcttagggaa tttgtgttta agaatattga\n",
      "    22081 tggttatttt aaaatatatt ctaagcacac gcctattaat ttagagcgtg atctccctca\n",
      "    22141 gggtttttcg gctttagaac cattggtaga tttgccaata ggtattaaca tcactaggtt\n",
      "    22201 tcaaacttta cttgctttac atagaagtta tttgactcct gttgattctt cttcaggttg\n",
      "    22261 gacagctggt gctgcagctt attatgtggg ttatcttcaa cctaggactt ttctattaaa\n",
      "    22321 atataatgaa aatggaacca ttacagatgc tgtagactgt gcacttgacc ctctctcaga\n",
      "    22381 aacaaagtgt acgttgaact ccttcactgt agaaaaagga atctatcaaa cttctaactt\n",
      "    22441 tagagtccaa ccaacagaat ctattgttag atttcctaat attacaaact tgtgcccttt\n",
      "    22501 tcatgaagtt tttaacgcca ccacatttgc atctgtttat gcttggaaca ggaagagaat\n",
      "    22561 cagcaactgt gttgctgatt attctgtcat atataatttc gcaccatttt tcgcttttaa\n",
      "    22621 gtgttatgga gtgtctccta ctaaattaaa tgatctctgc tttactaatg tctatgcaga\n",
      "    22681 ttcatttgta attagaggta atgaagtcag ccaaatcgct ccagggcaaa ctggaaatat\n",
      "    22741 tgctgattat aattataaat taccagatga ttttacaggc tgcgttatag cttggaattc\n",
      "    22801 taacaagctt gattctaagc ctagtggtaa ttataattac ctgtatagat tgtttaggaa\n",
      "    22861 gtctaagctc aaaccttttg agagagatat ttcaactgaa atctatcagg ccggtaacag\n",
      "    22921 accttgtaat ggtgttgcag gtcctaattg ttactctcct ttacaatcat atggtttccg\n",
      "    22981 acccacttat ggtgttggtc accaaccata cagagtagta gtactttctt ttgaacttct\n",
      "    23041 acatgcacca gcaactgttt gtggacctaa aaagtctact aatttggtta aaaacaaatg\n",
      "    23101 tgtcaatttc aacttcaatg gtttaacagg cacaggtgtt cttactgagt ctaacaaaaa\n",
      "    23161 gtttctgcct ttccaacaat ttggcagaga cattgctgac actactgatg ctgtccgtga\n",
      "    23221 tccacagaca cttgagattc ttgacattac accatgttct tttggtggtg tcagtgttat\n",
      "    23281 aacaccagga acaaatactt ctaaccaggt tgctgttctt tatcagggtg ttaactgcac\n",
      "    23341 agaagtccct gttgctattc atgcagatca acttactcct acttggcgtg tttattctac\n",
      "    23401 aggttctaat gtttttcaaa cacgtgcagg ctgtttaata ggggctgaat atgtcaacaa\n",
      "    23461 ctcatatgag tgtgacatac ccattggtgc aggtatatgc gctagttatc agactcagac\n",
      "    23521 taagtctcat cggcgggcac gtagtgtagc tagtcaatcc atcattgcct acactatgtc\n",
      "    23581 acttggtgca gaaaattcag ttgcttactc taataactct attgccatac ccacaaattt\n",
      "    23641 tactattagt gttaccacag aaattctacc agtgtctatg accaagacat cagtagattg\n",
      "    23701 tacaatgtac atttgtggtg attcaactga atgcagcaat cttttgttgc aatatggcag\n",
      "    23761 tttttgtaca caattaaaac gtgctttaac tggaatagct gttgaacaag acaaaaacac\n",
      "    23821 ccaagaagtt tttgcacaag tcaaacaaat ttacaaaaca ccaccaatta aatattttgg\n",
      "    23881 tggttttaat ttttcacaaa tattaccaga tccatcaaaa ccaagcaaga ggtcatttat\n",
      "    23941 tgaagatcta cttttcaaca aagtgacact tgcagatgct ggcttcatca aacaatatgg\n",
      "    24001 tgattgcctt ggtgatattg ctgctagaga cctcatttgt gcacaaaagt ttaacggcct\n",
      "    24061 tactgttttg ccacctttgc tcacagatga aatgattgct caatacactt ctgcactgtt\n",
      "    24121 agcgggtaca atcacttctg gttggacctt tggtgcaggt gctgcattac aaataccatt\n",
      "    24181 tgctatgcaa atggcttata ggtttaatgg tattggagtt acacagaatg ttctctatga\n",
      "    24241 gaaccaaaaa ttgattgcca accaatttaa tagtgctatt ggcaaaattc aagactcact\n",
      "    24301 ttcttccaca gcaagtgcac ttggaaaact tcaagatgtg gtcaaccata atgcacaagc\n",
      "    24361 tttaaacacg cttgttaaac aacttagctc caaatttggt gcaatttcaa gtgttttaaa\n",
      "    24421 tgatatcctt tcacgtcttg acaaagttga ggctgaagtg caaattgata ggttgatcac\n",
      "    24481 aggcagactt caaagtttgc agacatatgt gactcaacaa ttaattagag ctgcagaaat\n",
      "    24541 cagagcttct gctaatcttg ctgctactaa aatgtcagag tgtgtacttg gacaatcaaa\n",
      "    24601 aagagttgat ttttgtggaa agggctatca tcttatgtcc ttccctcagt cagcacctca\n",
      "    24661 tggtgtagtc ttcttgcatg tgacttatgt ccctgcacaa gaaaagaact tcacaactgc\n",
      "    24721 tcctgccatt tgtcatgatg gaaaagcaca ctttcctcgt gaaggtgtct ttgtttcaaa\n",
      "    24781 tggcacacac tggtttgtaa cacaaaggaa tttttatgaa ccacaaatca ttactacaga\n",
      "    24841 caacacattt gtgtctggta actgtgatgt tgtaatagga attgtcaaca acacagttta\n",
      "    24901 tgatcctttg caacctgaat tagattcatt taaggaggag ttagataaat attttaagaa\n",
      "    24961 tcatacatca ccagatgttg atttaggtga catctctggc attaatgctt cagttgtaaa\n",
      "    25021 cattcaaaaa gaaattgacc gcctcaatga ggttgccaag aatttaaatg aatctctcat\n",
      "    25081 cgatctccaa gaacttggaa agtatgagca gtatataaaa tggccatggt acatttggct\n",
      "    25141 aggttttata gctggcttga ttgccatagt aatggtgaca attatgcttt gctgtatgac\n",
      "    25201 cagttgctgt agttgtctca agggctgttg ttcttgtgga tcctgctgca aatttgatga\n",
      "    25261 agacgactct gagccagtgc tcaaaggagt caaattacat tacacataaa cgaacttatg\n",
      "    25321 gatttgttta tgagaatctt tacaattgga actgtaactt tgaagcaagg tgaaatcaag\n",
      "    25381 gatgctactc cttcagattt tgttcgcgct actgcaacga taccgataca agcctcactc\n",
      "    25441 cctttcggat ggcttattgt tggcgttgca cttcttgctg tttttcagag cgcttccaaa\n",
      "    25501 atcataactc tcaaaaagag atggcaacta gcactctcca agggtgttca ctttgtttgc\n",
      "    25561 aacttgctgt tgttgtttgt aacagtttac tcacaccttt tgctcgttgc tgctggcctt\n",
      "    25621 gaagcccctt ttctctatct ttatgcttta gtctacttct tgcagagtat aaactttgta\n",
      "    25681 agaataataa tgaggctttg gctttgctgg aaatgccgtt ccaaaaaccc attactttat\n",
      "    25741 gatgccaact attttctttg ctggcatact aattgttacg actattgtat accttacaat\n",
      "    25801 agtgtaactt cttcaattgt cattacttca ggtgatggca caacaagtcc tatttctgaa\n",
      "    25861 catgactacc agattggtgg ttatactgaa aaatgggaat ctggagtaaa agactgtgtt\n",
      "    25921 gtattacaca gttacttcac ttcagactat taccagctgt actcaactca attgagtaca\n",
      "    25981 gacattggtg ttgaacatgt taccttcttc atctacaata aaattgttga tgagcctgaa\n",
      "    26041 gaacatgtcc aaattcacac aatcgacggt tcatccggag ttgttaatcc agtaatggaa\n",
      "    26101 ccaatttatg atgaaccgac gacgactact agcgtgcctt tgtaagcaca agctgatgag\n",
      "    26161 tacgaactta tgtactcatt cgtttcggaa gagataggtg cgttaatagt taatagcgta\n",
      "    26221 cttctttttc ttgctttcgt ggtattcttg ctagttacac tagccatcct tactgcgctt\n",
      "    26281 cgattgtgtg cgtactgctg caatattgtt aacgtgagtc ttgtaaaacc ttctttttac\n",
      "    26341 gtttactctc gtgttaaaaa tctgaattct tctagagttc ctgatcttct ggtctaaacg\n",
      "    26401 aactaaatat tatattagtt tttctgtttg gaactttaat tttagccatg gcagattcca\n",
      "    26461 acggtactat taccgttgaa gagcttaaaa agctccttga agaatggaac ctagtaatag\n",
      "    26521 gtttcctatt ccttacatgg atttgtcttc tacaatttgc ctatgccaat aggaataggt\n",
      "    26581 ttttgtatat aattaagtta attttcctct ggctgttatg gccagtaact ttaacttgtt\n",
      "    26641 ttgtgcttgc tgctgtttac agaataaatt ggatcaccgg tggaattgct atcgcaatgg\n",
      "    26701 cttgtcttgt aggcttgatg tggctcagct acttcattgc ttctttcaga ctgtttgcgc\n",
      "    26761 gtacgcgttc catgtggtca tttaatccag aaactaacat tcttctcaac gtgccactcc\n",
      "    26821 atggcactat tctgaccaga ccgcttctag aaagtgaact cgtaatcgga gctgtgatcc\n",
      "    26881 ttcgtggaca tcttcgtatt gctggacacc atctaggacg ctgtgacatc aaggacctgc\n",
      "    26941 ctaaagaaat cactgttgct acatcacgaa cgctttctta ttacaaattg ggagcttcgc\n",
      "    27001 agcgtgtagc aggtgactca ggttttgctg catacagtcg ctacaggatt ggcaactata\n",
      "    27061 aattaaacac agaccattcc agtagcagtg acaatattgc tttgcttgta cagtaagtga\n",
      "    27121 caacagatgt ttcatctcgt tgactttcag gttactatag cagagatatt actaattatt\n",
      "    27181 atgcggactt ttaaagtttc catttggaat cttgattaca tcataaacct cataattaaa\n",
      "    27241 aatttatcta agtcactaac tgagaataaa tattctcaat tagatgaaga gcaaccaatg\n",
      "    27301 gagattctct aaacgaacat gaaaattatt cttttcttgg cactgataac actcgctact\n",
      "    27361 tgtgagcttt atcactacca agagtgtgtt agaggtacaa cagtactttt aaaagaacct\n",
      "    27421 tgctcttctg gaacatacga gggcaattca ccatttcatc ctctagctga taacaaattt\n",
      "    27481 gcactgactt gctttagcac tcaatttgct tttgcttgtc ctgacggcgt aaaacacgtc\n",
      "    27541 tatcagttac gtgccagatc agtttcacct aaactgttca tcagacaaga ggaagttcaa\n",
      "    27601 gaactttact ctccaatttt tcttattgtt gcggcaatag tgtttataac actttgcttc\n",
      "    27661 acactcaaaa gaaagacaga atgattgaac tttcattaat tgacttctat ttgtgctttt\n",
      "    27721 tagcctttct gttattcctt gttttaatta tgcttattat cttttggttc tcacttgaac\n",
      "    27781 tgcaagatca taatgaaact tgtcacgcct aaacgaacat gaaatttctt gttttcttat\n",
      "    27841 gaatcatcac aactgtagct gcatttcacc aagaatgtag tttacagtca tgtactcaac\n",
      "    27901 atcaaccata tgtagttgat gacccgtgtc ctattcactt ctattctaaa tggtatatta\n",
      "    27961 gagtaggagc tagaaaatca gcacctttaa ttgaattgtg cgtggatgag gctggttcta\n",
      "    28021 aatcacccat tcagtacatc gatatcggta attatacagt ttcctgttta ccttttacaa\n",
      "    28081 ttaattgcca ggaacctaaa ttgggtagtc ttgtagtgcg ttgttcgttc tatgaagact\n",
      "    28141 ttttagagta tcatgacgtt cgtgttgttt tagatttcat ctaaacgaac aaacttaaat\n",
      "    28201 gtctgataat ggaccccaaa accagcgaaa tgcactccgc attacgtttg gtggaccctc\n",
      "    28261 agattcaact ggcagtaacc agaatggtgg ggcgcgatca aaacaacgtc ggccccaagg\n",
      "    28321 tttacccaat aatactgcgt cttggttcac cgctctcact cagcatggca aggaagacct\n",
      "    28381 taaattccct cgaggacaag gcgttccaat taacaccaat agcagtccag atgaccaaat\n",
      "    28441 tggctactac cgaagagcta ccagacgaat tcgtggtggt gacggtaaaa tgaaagatct\n",
      "    28501 cagtccaaga tggtatttct actacctagg aactgggcca gaagctggac ttccctatgg\n",
      "    28561 tgctaacaaa gacggcatca tatgggttgc aactgaggga gccttgaata caccaaaaga\n",
      "    28621 tcacattggc acccgcaatc ctgctaacaa tgctgcaatc gtgctacaac ttcctcaagg\n",
      "    28681 aacaacattg ccaaaaggct tctacgcaga agggagcaga ggcggcagtc aagcctcttc\n",
      "    28741 tcgttcctca tcacgtagtc gcaacagttc aagaaattca actccaggca acagtaaacg\n",
      "    28801 aacttctcct gctagaatgg ctggcaatgg cggtgatgct gctcttgctt tgctgctgct\n",
      "    28861 tgacagattg aaccagcttg agagcaaaat gtctggtaaa ggccaacaac aacaaggcca\n",
      "    28921 aactgtcact aagaaatctg ctgctgaggc ttctaagaag cctcggcaaa aacgtactgc\n",
      "    28981 cactaaagca tacaatgtaa cacaagcttt cggcagacgt ggtccagaac aaacccaagg\n",
      "    29041 aaattttggg gaccaggaac taatcagaca aggaactgat tacaaacatt ggccgcaaat\n",
      "    29101 tgcacaattt gcccccagcg cttcagcgtt cttcggaatg tcgcgcattg gcatggaagt\n",
      "    29161 cacaccttcg ggaacgtggt tgacctacac aggtgccatc aaattggatg acaaagatcc\n",
      "    29221 aaatttcaaa gatcaagtca ttttgctgaa taagcatatt gacgcataca aaacattccc\n",
      "    29281 accaacagag cctaaaaagg ataaaaagaa gaaggctgat gaaactcaag ccttaccgca\n",
      "    29341 gagacagaag aaacagcaaa ctgtgactct tcttcctgct gcagatttgg atgatttctc\n",
      "    29401 caaacaattg caacaatcca tgagccgtgc tgactcaact caggcctaaa ctcatgcaga\n",
      "    29461 ccacacaagg cagatgggct atataaacgt tttcgctttt ccgtttacga tatatagtct\n",
      "    29521 actcttgtgc agaatgaatt ctcgtaacta catagcacaa gtagatgtag ttaactttaa\n",
      "    29581 tctcacatag caatctttaa tcagtgtgta acattaggga ggacttgaaa gagccaccac\n",
      "    29641 attttcac\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.efetch(db=\"nucleotide\", id='2623675418', rettype=\"gb\", retmode=\"text\")\n",
    "print(handle.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = Entrez.esearch(db='nucleotide', term='accD[Gene Name] AND \"E. coli\"[Organism]', retmax=\"20\")\n",
    "result_list = Entrez.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['2125548482', '2125547083', '2125545051', '2125544115', '2125543024', '2125541822', '2125536048', '2125532027', '2125529603', '2125527650', '2125514834', '2125511716', '2125507780', '2125503277', '2125499772', '2125499237', '2125493107', '2125481180', '2125479731', '2125477219']\n",
      "\n",
      "\n",
      "246849\n"
     ]
    }
   ],
   "source": [
    "id_list = result_list['IdList']\n",
    "count = result_list['Count']\n",
    "\n",
    "print(id_list)\n",
    "print(\"\\n\")\n",
    "print(count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. PDB"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.PDB import PDBParser,PDBList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on class PDBList in module Bio.PDB.PDBList:\n",
      "\n",
      "class PDBList(builtins.object)\n",
      " |  Quick access to the structure lists on the PDB or its mirrors.\n",
      " |  \n",
      " |  This class provides quick access to the structure lists on the\n",
      " |  PDB server or its mirrors. The structure lists contain\n",
      " |  four-letter PDB codes, indicating that structures are\n",
      " |  new, have been modified or are obsolete. The lists are released\n",
      " |  on a weekly basis.\n",
      " |  \n",
      " |  It also provides a function to retrieve PDB files from the server.\n",
      " |  To use it properly, prepare a directory /pdb or the like,\n",
      " |  where PDB files are stored.\n",
      " |  \n",
      " |  All available file formats (PDB, PDBx/mmCif, PDBML, mmtf) are supported.\n",
      " |  Please note that large structures (containing >62 chains\n",
      " |  and/or 99999 ATOM lines) are no longer stored as a single PDB file\n",
      " |  and by default (when PDB format selected) are not downloaded.\n",
      " |  \n",
      " |  Large structures can be downloaded in other formats, including PDBx/mmCif\n",
      " |  or as a .tar file (a collection of PDB-like formatted files for a given\n",
      " |  structure).\n",
      " |  \n",
      " |  If you want to use this module from inside a proxy, add\n",
      " |  the proxy variable to your environment, e.g. in Unix:\n",
      " |  export HTTP_PROXY='http://realproxy.charite.de:888'\n",
      " |  (This can also be added to ~/.bashrc)\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __init__(self, server='ftp://ftp.wwpdb.org', pdb=None, obsolete_pdb=None, verbose=True)\n",
      " |      Initialize the class with the default server or a custom one.\n",
      " |      \n",
      " |      Argument pdb is the local path to use, defaulting to the current\n",
      " |      directory at the moment of initialisation.\n",
      " |  \n",
      " |  download_entire_pdb(self, listfile=None, file_format=None)\n",
      " |      Retrieve all PDB entries not present in the local PDB copy.\n",
      " |      \n",
      " |      :param listfile: filename to which all PDB codes will be written (optional)\n",
      " |      \n",
      " |      :param file_format:\n",
      " |          File format. Available options:\n",
      " |      \n",
      " |          * \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          * \"pdb\" (format PDB),\n",
      " |          * \"xml\" (PMDML/XML format),\n",
      " |          * \"mmtf\" (highly compressed),\n",
      " |          * \"bundle\" (PDB formatted archive for large structure}\n",
      " |      \n",
      " |      NOTE. The default download format has changed from PDB to PDBx/mmCif\n",
      " |  \n",
      " |  download_obsolete_entries(self, listfile=None, file_format=None)\n",
      " |      Retrieve all obsolete PDB entries not present in local obsolete PDB copy.\n",
      " |      \n",
      " |      :param listfile: filename to which all PDB codes will be written (optional)\n",
      " |      \n",
      " |      :param file_format: file format. Available options:\n",
      " |          \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          \"pdb\" (format PDB),\n",
      " |          \"xml\" (PMDML/XML format),\n",
      " |      \n",
      " |      NOTE. The default download format has changed from PDB to PDBx/mmCif\n",
      " |  \n",
      " |  download_pdb_files(self, pdb_codes, obsolete=False, pdir=None, file_format=None, overwrite=False)\n",
      " |      Fetch set of PDB structure files from the PDB server and stores them locally.\n",
      " |      \n",
      " |      The PDB structure's file name is returned as a single string.\n",
      " |      If obsolete ``==`` True, the files will be saved in a special file tree.\n",
      " |      \n",
      " |      :param pdb_codes: a list of 4-symbols structure Ids from PDB\n",
      " |      :type pdb_codes: list of strings\n",
      " |      \n",
      " |      :param file_format:\n",
      " |          File format. Available options:\n",
      " |      \n",
      " |          * \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          * \"pdb\" (format PDB),\n",
      " |          * \"xml\" (PMDML/XML format),\n",
      " |          * \"mmtf\" (highly compressed),\n",
      " |          * \"bundle\" (PDB formatted archive for large structure}\n",
      " |      \n",
      " |      :param overwrite: if set to True, existing structure files will be overwritten. Default: False\n",
      " |      :type overwrite: bool\n",
      " |      \n",
      " |      :param obsolete:\n",
      " |          Has a meaning only for obsolete structures.\n",
      " |          If True, download the obsolete structure\n",
      " |          to 'obsolete' folder, otherwise download won't be performed.\n",
      " |          This option doesn't work for mmtf format as obsoleted structures are not availbe as mmtf.\n",
      " |          (default: False)\n",
      " |      \n",
      " |      :type obsolete: bool\n",
      " |      \n",
      " |      :param pdir: put the file in this directory (default: create a PDB-style directory tree)\n",
      " |      :type pdir: string\n",
      " |      \n",
      " |      :return: filenames\n",
      " |      :rtype: string\n",
      " |  \n",
      " |  get_all_entries(self)\n",
      " |      Retrieve the big file containing all the PDB entries and some annotation.\n",
      " |      \n",
      " |      Returns a list of PDB codes in the index file.\n",
      " |  \n",
      " |  get_all_obsolete(self)\n",
      " |      Return a list of all obsolete entries ever in the PDB.\n",
      " |      \n",
      " |      Returns a list of all obsolete pdb codes that have ever been\n",
      " |      in the PDB.\n",
      " |      \n",
      " |      Gets and parses the file from the PDB server in the format\n",
      " |      (the first pdb_code column is the one used). The file looks\n",
      " |      like this::\n",
      " |      \n",
      " |           LIST OF OBSOLETE COORDINATE ENTRIES AND SUCCESSORS\n",
      " |          OBSLTE    31-JUL-94 116L     216L\n",
      " |          ...\n",
      " |          OBSLTE    29-JAN-96 1HFT     2HFT\n",
      " |          OBSLTE    21-SEP-06 1HFV     2J5X\n",
      " |          OBSLTE    21-NOV-03 1HG6\n",
      " |          OBSLTE    18-JUL-84 1HHB     2HHB 3HHB\n",
      " |          OBSLTE    08-NOV-96 1HID     2HID\n",
      " |          OBSLTE    01-APR-97 1HIU     2HIU\n",
      " |          OBSLTE    14-JAN-04 1HKE     1UUZ\n",
      " |          ...\n",
      " |  \n",
      " |  get_recent_changes(self)\n",
      " |      Return three lists of the newest weekly files (added,mod,obsolete).\n",
      " |      \n",
      " |      Reads the directories with changed entries from the PDB server and\n",
      " |      returns a tuple of three URL's to the files of new, modified and\n",
      " |      obsolete entries from the most recent list. The directory with the\n",
      " |      largest numerical name is used.\n",
      " |      Returns None if something goes wrong.\n",
      " |      \n",
      " |      Contents of the data/status dir (20031013 would be used);:\n",
      " |      \n",
      " |          drwxrwxr-x   2 1002     sysadmin     512 Oct  6 18:28 20031006\n",
      " |          drwxrwxr-x   2 1002     sysadmin     512 Oct 14 02:14 20031013\n",
      " |          -rw-r--r--   1 1002     sysadmin    1327 Mar 12  2001 README\n",
      " |  \n",
      " |  get_seqres_file(self, savefile='pdb_seqres.txt')\n",
      " |      Retrieve and save a (big) file containing all the sequences of PDB entries.\n",
      " |  \n",
      " |  retrieve_pdb_file(self, pdb_code, obsolete=False, pdir=None, file_format=None, overwrite=False)\n",
      " |      Fetch PDB structure file from PDB server, and store it locally.\n",
      " |      \n",
      " |      The PDB structure's file name is returned as a single string.\n",
      " |      If obsolete ``==`` True, the file will be saved in a special file tree.\n",
      " |      \n",
      " |      NOTE. The default download format has changed from PDB to PDBx/mmCif\n",
      " |      \n",
      " |      :param pdb_code: 4-symbols structure Id from PDB (e.g. 3J92).\n",
      " |      :type pdb_code: string\n",
      " |      \n",
      " |      :param file_format:\n",
      " |          File format. Available options:\n",
      " |      \n",
      " |          * \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          * \"pdb\" (format PDB),\n",
      " |          * \"xml\" (PDBML/XML format),\n",
      " |          * \"mmtf\" (highly compressed),\n",
      " |          * \"bundle\" (PDB formatted archive for large structure}\n",
      " |      \n",
      " |      :type file_format: string\n",
      " |      \n",
      " |      :param overwrite: if set to True, existing structure files will be overwritten. Default: False\n",
      " |      :type overwrite: bool\n",
      " |      \n",
      " |      :param obsolete:\n",
      " |          Has a meaning only for obsolete structures. If True, download the obsolete structure\n",
      " |          to 'obsolete' folder, otherwise download won't be performed.\n",
      " |          This option doesn't work for mmtf format as obsoleted structures aren't stored in mmtf.\n",
      " |          Also doesn't have meaning when parameter pdir is specified.\n",
      " |          Note: make sure that you are about to download the really obsolete structure.\n",
      " |          Trying to download non-obsolete structure into obsolete folder will not work\n",
      " |          and you face the \"structure doesn't exists\" error.\n",
      " |          Default: False\n",
      " |      \n",
      " |      :type obsolete: bool\n",
      " |      \n",
      " |      :param pdir: put the file in this directory (default: create a PDB-style directory tree)\n",
      " |      :type pdir: string\n",
      " |      \n",
      " |      :return: filename\n",
      " |      :rtype: string\n",
      " |  \n",
      " |  update_pdb(self, file_format=None)\n",
      " |      Update your local copy of the PDB files.\n",
      " |      \n",
      " |      I guess this is the 'most wanted' function from this module.\n",
      " |      It gets the weekly lists of new and modified pdb entries and\n",
      " |      automatically downloads the according PDB files.\n",
      " |      You can call this module as a weekly cron job.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Static methods defined here:\n",
      " |  \n",
      " |  get_status_list(url)\n",
      " |      Retrieve a list of pdb codes in the weekly pdb status file from given URL.\n",
      " |      \n",
      " |      Used by get_recent_changes. Typical contents of the list files parsed\n",
      " |      by this method is now very simply - one PDB name per line.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes defined here:\n",
      " |  \n",
      " |  PDB_REF = '\\n    The Protein Data Bank: a computer-based arc... pp. 53...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(PDBList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading PDB structure '7BYR'...\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'dir/pdb7byr.ent'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdbl=PDBList()\n",
    "pdbl.retrieve_pdb_file(\"7BYR\", file_format=\"pdb\", pdir=\"dir\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rhyme/.local/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 26237.\n",
      "  PDBConstructionWarning,\n",
      "/home/rhyme/.local/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 26405.\n",
      "  PDBConstructionWarning,\n",
      "/home/rhyme/.local/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain C is discontinuous at line 26545.\n",
      "  PDBConstructionWarning,\n"
     ]
    }
   ],
   "source": [
    "parser = PDBParser()\n",
    "structure = parser.get_structure(\"7BYR\",\"dir/pdb7byr.ent\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chainid: A\n",
      "chainid: B\n",
      "chainid: C\n",
      "chainid: H\n",
      "chainid: L\n",
      "chainid: D\n",
      "chainid: E\n",
      "chainid: F\n",
      "chainid: G\n",
      "chainid: I\n",
      "chainid: J\n"
     ]
    }
   ],
   "source": [
    "for chain in structure[0]:\n",
    "    print(f\"chainid: {chain.id}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3.84"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resolution= structure.header[\"resolution\"]\n",
    "resolution #angstrom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'sars-cov-2, antigen, rbd, neutralizing antibody, viral protein'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keywords = structure.header[\"keywords\"]\n",
    "keywords"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. EXPASY"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.1. PROSITE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import ExPASy\n",
    "from Bio.ExPASy import Prosite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on module Bio.ExPASy.Prosite in Bio.ExPASy:\n",
      "\n",
      "NAME\n",
      "    Bio.ExPASy.Prosite - Parser for the prosite dat file from Prosite at ExPASy.\n",
      "\n",
      "DESCRIPTION\n",
      "    See https://www.expasy.org/prosite/\n",
      "    \n",
      "    Tested with:\n",
      "     - Release 20.43, 10-Feb-2009\n",
      "     - Release 2017_03 of 15-Mar-2017.\n",
      "    \n",
      "    Functions:\n",
      "     - read                  Reads a Prosite file containing one Prosite record\n",
      "     - parse                 Iterates over records in a Prosite file.\n",
      "    \n",
      "    Classes:\n",
      "     - Record                Holds Prosite data.\n",
      "\n",
      "CLASSES\n",
      "    builtins.object\n",
      "        Record\n",
      "    \n",
      "    class Record(builtins.object)\n",
      "     |  Holds information from a Prosite record.\n",
      "     |  \n",
      "     |  Main attributes:\n",
      "     |   - name           ID of the record.  e.g. ADH_ZINC\n",
      "     |   - type           Type of entry.  e.g. PATTERN, MATRIX, or RULE\n",
      "     |   - accession      e.g. PS00387\n",
      "     |   - created        Date the entry was created.  (MMM-YYYY for releases\n",
      "     |     before January 2017, DD-MMM-YYYY since January 2017)\n",
      "     |   - data_update    Date the 'primary' data was last updated.\n",
      "     |   - info_update    Date data other than 'primary' data was last updated.\n",
      "     |   - pdoc           ID of the PROSITE DOCumentation.\n",
      "     |   - description    Free-format description.\n",
      "     |   - pattern        The PROSITE pattern.  See docs.\n",
      "     |   - matrix         List of strings that describes a matrix entry.\n",
      "     |   - rules          List of rule definitions (from RU lines).  (strings)\n",
      "     |   - prorules       List of prorules (from PR lines). (strings)\n",
      "     |  \n",
      "     |  NUMERICAL RESULTS:\n",
      "     |   - nr_sp_release  SwissProt release.\n",
      "     |   - nr_sp_seqs     Number of seqs in that release of Swiss-Prot. (int)\n",
      "     |   - nr_total       Number of hits in Swiss-Prot.  tuple of (hits, seqs)\n",
      "     |   - nr_positive    True positives.  tuple of (hits, seqs)\n",
      "     |   - nr_unknown     Could be positives.  tuple of (hits, seqs)\n",
      "     |   - nr_false_pos   False positives.  tuple of (hits, seqs)\n",
      "     |   - nr_false_neg   False negatives.  (int)\n",
      "     |   - nr_partial     False negatives, because they are fragments. (int)\n",
      "     |  \n",
      "     |  COMMENTS:\n",
      "     |   - cc_taxo_range  Taxonomic range.  See docs for format\n",
      "     |   - cc_max_repeat  Maximum number of repetitions in a protein\n",
      "     |   - cc_site        Interesting site.  list of tuples (pattern pos, desc.)\n",
      "     |   - cc_skip_flag   Can this entry be ignored?\n",
      "     |   - cc_matrix_type\n",
      "     |   - cc_scaling_db\n",
      "     |   - cc_author\n",
      "     |   - cc_ft_key\n",
      "     |   - cc_ft_desc\n",
      "     |   - cc_version     version number (introduced in release 19.0)\n",
      "     |  \n",
      "     |  The following are all lists if tuples (swiss-prot accession, swiss-prot name).\n",
      "     |  \n",
      "     |  DATA BANK REFERENCES:\n",
      "     |   - dr_positive\n",
      "     |   - dr_false_neg\n",
      "     |   - dr_false_pos\n",
      "     |   - dr_potential   Potential hits, but fingerprint region not yet available.\n",
      "     |   - dr_unknown     Could possibly belong\n",
      "     |   - pdb_structs    List of PDB entries.\n",
      "     |  \n",
      "     |  Methods defined here:\n",
      "     |  \n",
      "     |  __init__(self)\n",
      "     |      Initialize the class.\n",
      "     |  \n",
      "     |  ----------------------------------------------------------------------\n",
      "     |  Data descriptors defined here:\n",
      "     |  \n",
      "     |  __dict__\n",
      "     |      dictionary for instance variables (if defined)\n",
      "     |  \n",
      "     |  __weakref__\n",
      "     |      list of weak references to the object (if defined)\n",
      "\n",
      "FUNCTIONS\n",
      "    parse(handle)\n",
      "        Parse Prosite records.\n",
      "        \n",
      "        This function is for parsing Prosite files containing multiple\n",
      "        records.\n",
      "        \n",
      "        Arguments:\n",
      "         - handle   - handle to the file.\n",
      "    \n",
      "    read(handle)\n",
      "        Read one Prosite record.\n",
      "        \n",
      "        This function is for parsing Prosite files containing\n",
      "        exactly one record.\n",
      "        \n",
      "        Arguments:\n",
      "         - handle   - handle to the file.\n",
      "\n",
      "FILE\n",
      "    /home/rhyme/.local/lib/python3.6/site-packages/Bio/ExPASy/Prosite.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(Prosite)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = ExPASy.get_prosite_raw('PS51442')\n",
    "record = Prosite.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Coronavirus main protease (M-pro) domain profile.\n"
     ]
    }
   ],
   "source": [
    "print(record.description)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1LVO', '1P9S', '1P9U', '1Q2W', '1UJ1', '1UK2', '1UK3', '1UK4', '1WOF', '1Z1I']\n"
     ]
    }
   ],
   "source": [
    "print(record.pdb_structs[:10]) #so these are 10 proteins of domain profile coronavirus main protease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N-{P}-[ST]-{P}.\n"
     ]
    }
   ],
   "source": [
    "handle = ExPASy.get_prosite_raw('PS00001')\n",
    "record = Prosite.read(handle)\n",
    "print(record.pattern) #the common pattern within domain is asparagine, proline, serine and threonine"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.2. ScanProsite"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.ExPASy import ScanProsite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "258"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_record = SeqIO.read(\"prot_seq.fasta\", format=\"fasta\")\n",
    "len(prot_record.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "ename": "HTTPError",
     "evalue": "HTTP Error 308: Permanent Redirect",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-29-8b263e207bef>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mScanProsite\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscan\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mprot_record\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmirror\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"https://prosite.expasy.org/\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mScanProsite\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/Bio/ExPASy/ScanProsite.py\u001b[0m in \u001b[0;36mscan\u001b[0;34m(seq, mirror, output, **keywords)\u001b[0m\n\u001b[1;32m     57\u001b[0m     \u001b[0mcommand\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murlencode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparameters\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     58\u001b[0m     \u001b[0murl\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"%s/cgi-bin/prosite/PSScan.cgi?%s\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmirror\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcommand\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 59\u001b[0;31m     \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     60\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mhandle\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     61\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[1;32m    221\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m         \u001b[0mopener\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_opener\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 223\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mopener\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    224\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    225\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0minstall_opener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mopener\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mopen\u001b[0;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[1;32m    530\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mprocessor\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprocess_response\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprotocol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    531\u001b[0m             \u001b[0mmeth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprocessor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 532\u001b[0;31m             \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmeth\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    533\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    534\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mhttp_response\u001b[0;34m(self, request, response)\u001b[0m\n\u001b[1;32m    640\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;36m200\u001b[0m \u001b[0;34m<=\u001b[0m \u001b[0mcode\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m300\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    641\u001b[0m             response = self.parent.error(\n\u001b[0;32m--> 642\u001b[0;31m                 'http', request, response, code, msg, hdrs)\n\u001b[0m\u001b[1;32m    643\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    644\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36merror\u001b[0;34m(self, proto, *args)\u001b[0m\n\u001b[1;32m    568\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mhttp_err\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    569\u001b[0m             \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'default'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'http_error_default'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0morig_args\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 570\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_chain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    571\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    572\u001b[0m \u001b[0;31m# XXX probably also want an abstract factory that knows when it makes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36m_call_chain\u001b[0;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[1;32m    502\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mhandler\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mhandlers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    503\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 504\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    505\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    506\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mhttp_error_default\u001b[0;34m(self, req, fp, code, msg, hdrs)\u001b[0m\n\u001b[1;32m    648\u001b[0m \u001b[0;32mclass\u001b[0m \u001b[0mHTTPDefaultErrorHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    649\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mhttp_error_default\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 650\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mHTTPError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfull_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    651\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    652\u001b[0m \u001b[0;32mclass\u001b[0m \u001b[0mHTTPRedirectHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mHTTPError\u001b[0m: HTTP Error 308: Permanent Redirect"
     ]
    }
   ],
   "source": [
    "handle = ScanProsite.scan(seq=prot_record.seq, mirror=\"https://prosite.expasy.org/\")\n",
    "result = ScanProsite.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'result' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-30-39948d895def>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mn_match\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'result' is not defined"
     ]
    }
   ],
   "source": [
    "result.n_match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'result' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-31-2c43f6f8ed53>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mresult\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'result' is not defined"
     ]
    }
   ],
   "source": [
    "result[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5. KEGG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.KEGG import REST, Enzyme #pathways and genes that associated with KEGG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on package Bio.KEGG.Enzyme in Bio.KEGG:\n",
      "\n",
      "NAME\n",
      "    Bio.KEGG.Enzyme - Code to work with the KEGG Enzyme database.\n",
      "\n",
      "DESCRIPTION\n",
      "    Functions:\n",
      "     - parse - Returns an iterator giving Record objects.\n",
      "    \n",
      "    Classes:\n",
      "     - Record - Holds the information from a KEGG Enzyme record.\n",
      "\n",
      "PACKAGE CONTENTS\n",
      "\n",
      "\n",
      "CLASSES\n",
      "    builtins.object\n",
      "        Record\n",
      "    \n",
      "    class Record(builtins.object)\n",
      "     |  Holds info from a KEGG Enzyme record.\n",
      "     |  \n",
      "     |  Attributes:\n",
      "     |   - entry       The EC number (withou the 'EC ').\n",
      "     |   - name        A list of the enzyme names.\n",
      "     |   - classname   A list of the classification terms.\n",
      "     |   - sysname     The systematic name of the enzyme.\n",
      "     |   - reaction    A list of the reaction description strings.\n",
      "     |   - substrate   A list of the substrates.\n",
      "     |   - product     A list of the products.\n",
      "     |   - inhibitor   A list of the inhibitors.\n",
      "     |   - cofactor    A list of the cofactors.\n",
      "     |   - effector    A list of the effectors.\n",
      "     |   - comment     A list of the comment strings.\n",
      "     |   - pathway     A list of 3-tuples: (database, id, pathway)\n",
      "     |   - genes       A list of 2-tuples: (organism, list of gene ids)\n",
      "     |   - disease     A list of 3-tuples: (database, id, disease)\n",
      "     |   - structures  A list of 2-tuples: (database, list of struct ids)\n",
      "     |   - dblinks     A list of 2-tuples: (database, list of db ids)\n",
      "     |  \n",
      "     |  Methods defined here:\n",
      "     |  \n",
      "     |  __init__(self)\n",
      "     |      Initialize a new Record.\n",
      "     |  \n",
      "     |  __str__(self)\n",
      "     |      Return a string representation of this Record.\n",
      "     |  \n",
      "     |  ----------------------------------------------------------------------\n",
      "     |  Data descriptors defined here:\n",
      "     |  \n",
      "     |  __dict__\n",
      "     |      dictionary for instance variables (if defined)\n",
      "     |  \n",
      "     |  __weakref__\n",
      "     |      list of weak references to the object (if defined)\n",
      "\n",
      "FUNCTIONS\n",
      "    parse(handle)\n",
      "        Parse a KEGG Enzyme file, returning Record objects.\n",
      "        \n",
      "        This is an iterator function, typically used in a for loop.  For\n",
      "        example, using one of the example KEGG files in the Biopython\n",
      "        test suite,\n",
      "        \n",
      "        >>> with open(\"KEGG/enzyme.sample\") as handle:\n",
      "        ...     for record in parse(handle):\n",
      "        ...         print(\"%s %s\" % (record.entry, record.name[0]))\n",
      "        ...\n",
      "        1.1.1.1 alcohol dehydrogenase\n",
      "        1.1.1.62 17beta-estradiol 17-dehydrogenase\n",
      "        1.1.1.68 Transferred to 1.5.1.20\n",
      "        1.6.5.3 NADH:ubiquinone reductase (H+-translocating)\n",
      "        1.14.13.28 3,9-dihydroxypterocarpan 6a-monooxygenase\n",
      "        2.4.1.68 glycoprotein 6-alpha-L-fucosyltransferase\n",
      "        3.1.1.6 acetylesterase\n",
      "        2.7.2.1 acetate kinase\n",
      "    \n",
      "    read(handle)\n",
      "        Parse a KEGG Enzyme file with exactly one entry.\n",
      "        \n",
      "        If the handle contains no records, or more than one record,\n",
      "        an exception is raised.  For example:\n",
      "        \n",
      "        >>> with open(\"KEGG/enzyme.new\") as handle:\n",
      "        ...     record = read(handle)\n",
      "        ...     print(\"%s %s\" % (record.entry, record.name[0]))\n",
      "        ...\n",
      "        6.2.1.25 benzoate---CoA ligase\n",
      "\n",
      "DATA\n",
      "    name_wrap = [0, '', (' ', '$', 1, 1), ('-', '$', 1, 1)]\n",
      "    rxn_wrap = [0, '', (' + ', '', 1, 1), (' = ', '', 1, 1), (' ', '$', 1,...\n",
      "\n",
      "FILE\n",
      "    /home/rhyme/.local/lib/python3.6/site-packages/Bio/KEGG/Enzyme/__init__.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(Enzyme)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "274949"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "request = REST.kegg_get(\"ec:5.4.2.2\") #nzymecommission number\n",
    "open(\"ec_5.4.2.2.txt\", \"w\").write(request.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Isomerases;',\n",
       " 'Intramolecular transferases;',\n",
       " 'Phosphotransferases (phosphomutases)']"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "records = Enzyme.parse(open(\"ec_5.4.2.2.txt\"))\n",
    "record = list(records)[0]\n",
    "record.classname"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('PATH', 'ec00010', 'Glycolysis / Gluconeogenesis'),\n",
       " ('PATH', 'ec00030', 'Pentose phosphate pathway'),\n",
       " ('PATH', 'ec00052', 'Galactose metabolism'),\n",
       " ('PATH', 'ec00230', 'Purine metabolism'),\n",
       " ('PATH', 'ec00500', 'Starch and sucrose metabolism'),\n",
       " ('PATH', 'ec00520', 'Amino sugar and nucleotide sugar metabolism'),\n",
       " ('PATH', 'ec00521', 'Streptomycin biosynthesis'),\n",
       " ('PATH', 'ec01100', 'Metabolic pathways'),\n",
       " ('PATH', 'ec01110', 'Biosynthesis of secondary metabolites'),\n",
       " ('PATH', 'ec01120', 'Microbial metabolism in diverse environments')]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.pathway  #all the pathways that associated with this enzyme"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('HSA', ['5236', '55276']),\n",
       " ('PTR', ['456908', '461162']),\n",
       " ('PPS', ['100977295', '100993927']),\n",
       " ('GGO', ['101128874', '101131551']),\n",
       " ('PON', ['100190836', '100438793']),\n",
       " ('NLE', ['100596081', '100600656']),\n",
       " ('HMH', ['116456694', '116457795']),\n",
       " ('MCC', ['100424648', '699401']),\n",
       " ('MCF', ['101925921', '102130622']),\n",
       " ('MTHB', ['126935012', '126954887'])]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.genes[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['HSA', 'PTR', 'PPS', 'GGO', 'PON', 'NLE', 'HMH', 'MCC', 'MCF', 'MTHB']\n"
     ]
    }
   ],
   "source": [
    "list_genes = []\n",
    "for x,y in record.genes:\n",
    "    list_genes += x.split(\"\\n\")\n",
    "    \n",
    "print(list_genes[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
